diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index 60e87e4183258..cba272c4b497e 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -14,6 +14,7 @@
 #include <string_view>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/debug/alias.h"
 #include "base/functional/bind.h"
 #include "base/i18n/rtl.h"
@@ -672,7 +673,11 @@ void Tab::OnMouseReleased(const ui::MouseEvent& event) {
   // Close tab on middle click, but only if the button is released over the tab
   // (normal windows behavior is to discard presses of a UI element where the
   // releases happen off the element).
-  if (event.IsOnlyMiddleMouseButton()) {
+  if (event.IsOnlyMiddleMouseButton() ||
+      // Close tab on double click, mirror of IsOnlyMiddleMouseButton
+      // Based on gz83's work.
+      ((event.IsOnlyLeftMouseButton() && event.GetClickCount() == 2) &&
+        base::CommandLine::ForCurrentProcess()->HasSwitch("double-click-close-tab"))) {
     if (HitTestPoint(event.location())) {
       controller_->CloseTab(this, CloseTabSource::kFromMouse);
     } else if (closing_) {
diff --git a/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc b/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
index eb9e13ce3d2dc..468c9d61b4ecb 100644
--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
@@ -7,6 +7,7 @@
 #include <optional>
 #include <string>
 
+#include "base/command_line.h"
 #include "base/functional/callback_helpers.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/user_metrics.h"
@@ -396,7 +397,12 @@ void VerticalTabView::OnMouseReleased(const ui::MouseEvent& event) {
 
   auto* controller = collection_node_->GetController();
   base::WeakPtr<VerticalTabView> self = weak_ptr_factory_.GetWeakPtr();
-  if (event.IsOnlyMiddleMouseButton()) {
+  if (event.IsOnlyMiddleMouseButton() ||
+      // Close tab on double click, mirror of IsOnlyMiddleMouseButton.
+      // Based on gz83's work.
+      (event.IsOnlyLeftMouseButton() && event.GetClickCount() == 2 &&
+       base::CommandLine::ForCurrentProcess()->HasSwitch(
+           "double-click-close-tab"))) {
     if (HitTestPoint(event.location())) {
       controller->CloseTab(GetTabInterface());
     }
