diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index e56f0ea663e4a..31ff6a4864bc6 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -1223,6 +1223,10 @@ void Tab::UpdateIconVisibility() {
     // Close button is shown on active tabs regardless of the size.
     showing_close_button_ = true;
 #endif  // BUILDFLAG(IS_CHROMEOS)
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons")) {
+      showing_close_button_ = false;
+    }
 
     if (features::IsTabStripDeclutterEnabled() && showing_close_button_) {
       const bool alert_fits_without_close =
@@ -1281,6 +1285,8 @@ void Tab::UpdateIconVisibility() {
         features::IsTabStripDeclutterEnabled() &&
         available_width <= TabStyle::kTabStripDeclutterMaxTabWidthForCloseHide;
     showing_close_button_ =
+        !base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons") &&
 #if BUILDFLAG(IS_CHROMEOS)
         should_show_close_button &&
 #endif
diff --git a/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc b/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
index b60e1eb76bc6c..0aeab1dcd4717 100644
--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
@@ -838,6 +838,11 @@ bool VerticalTabView::IsChildVisible(const views::View* child_view,
       return false;
     }
 
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons")) {
+      return false;
+    }
+
     // When uncollapsing the tabstrip, intentionally start showing the close
     // button on active non-hovered tabs a little bit sooner than reaching the
     // uncollapsed min width, because otherwise the close buttons in a grouped
