diff --git a/chrome/browser/download/chrome_download_manager_delegate.cc b/chrome/browser/download/chrome_download_manager_delegate.cc
index e7121d906cb41..10d160279d550 100644
--- a/chrome/browser/download/chrome_download_manager_delegate.cc
+++ b/chrome/browser/download/chrome_download_manager_delegate.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -2129,8 +2129,12 @@ bool ChromeDownloadManagerDelegate::IsOpenInBrowserPreferredForFile(
 bool ChromeDownloadManagerDelegate::ShouldBlockFile(
     download::DownloadItem* item,
     download::DownloadDangerType danger_type) const {
+  // Don't block downloads if flag is set.
+  static const bool allow_insecure_downloads =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "allow-insecure-downloads");
   // Chrome-initiated background downloads should not be blocked.
-  if (item && !item->RequireSafetyChecks()) {
+  if ((item && !item->RequireSafetyChecks()) || allow_insecure_downloads) {
     return false;
   }
 
@@ -2184,44 +2188,14 @@ bool ChromeDownloadManagerDelegate::ShouldBlockFile(
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadOpenedReport(
     DownloadItem* download,
     bool show_download_in_folder) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::DownloadProtectionService* service =
-      GetDownloadProtectionService();
-  if (service) {
-    service->MaybeSendDangerousDownloadOpenedReport(download,
-                                                    show_download_in_folder);
-  }
-#endif
+  return;
 }
 
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadCanceledReport(
     DownloadItem* download,
     bool is_shutdown) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::SafeBrowsingService* sb_service =
-      g_browser_process->safe_browsing_service();
-  if (!sb_service) {
-    return;
-  }
-  // Note: We cannot go through download_protection_service here, because this
-  // function may be called at shutdown. The download_protection_service
-  // object may already be deleted at this point.
-  if (is_shutdown) {
-    sb_service->PersistDownloadReportAndSendOnNextStartup(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_PROFILE_CLOSED,
-        /*did_proceed=*/false, std::nullopt);
-  } else {
-    sb_service->SendDownloadReport(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_AUTO_DELETED,
-        /*did_proceed=*/false, std::nullopt);
-  }
-#endif
+  return;
 }
-
 void ChromeDownloadManagerDelegate::CheckDownloadAllowed(
     const content::WebContents::Getter& web_contents_getter,
     const GURL& url,
diff --git a/chrome/browser/download/download_target_determiner.cc b/chrome/browser/download/download_target_determiner.cc
index 56b6e324e3a7a..a319ac415a8e2 100644
--- a/chrome/browser/download/download_target_determiner.cc
+++ b/chrome/browser/download/download_target_determiner.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -8,6 +8,7 @@
 #include <string>
 #include <vector>
 
+#include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
@@ -1236,6 +1237,14 @@ DownloadFileType::DangerLevel DownloadTargetDeterminer::GetDangerLevel(
     PriorVisitsToReferrer visits) const {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
 
+  static const bool allow_insecure_downloads =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "allow-insecure-downloads");
+  // Allow all downloads with this Thorium flag.
+  if (allow_insecure_downloads) {
+    return DownloadFileType::NOT_DANGEROUS;
+  }
+
   // User-initiated extension downloads from pref-whitelisted sources are not
   // considered dangerous.
   if (download_->HasUserGesture() &&
diff --git a/chrome/browser/download/insecure_download_blocking.cc b/chrome/browser/download/insecure_download_blocking.cc
index cfedddb9506df..c197bba515fac 100644
--- a/chrome/browser/download/insecure_download_blocking.cc
+++ b/chrome/browser/download/insecure_download_blocking.cc
@@ -1,4 +1,4 @@
-// Copyright 2019 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -6,6 +6,7 @@
 
 #include <optional>
 
+#include "base/command_line.h"
 #include "base/debug/crash_logging.h"
 #include "base/debug/dump_without_crashing.h"
 #include "base/memory/raw_ptr.h"
@@ -217,6 +218,9 @@ struct InsecureDownloadData {
     // used for Chrome stuff. Background fetch should probably be HTTPS-only.
     auto download_source = item->GetDownloadSource();
     auto transition_type = item->GetTransitionType();
+    static const bool allow_insecure_downloads =
+        base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "allow-insecure-downloads");
     if (download_source == DownloadSource::RETRY ||
         ui::PageTransitionCoreTypeIs(transition_type,
                                      ui::PAGE_TRANSITION_RELOAD) ||
@@ -232,7 +236,8 @@ struct InsecureDownloadData {
         download_source == DownloadSource::OFFLINE_PAGE ||
         download_source == DownloadSource::INTERNAL_API ||
         download_source == DownloadSource::EXTENSION_API ||
-        download_source == DownloadSource::EXTENSION_INSTALLER) {
+        download_source == DownloadSource::EXTENSION_INSTALLER ||
+        allow_insecure_downloads) {
       base::UmaHistogramEnumeration(
           kInsecureDownloadHistogramName,
           InsecureDownloadSecurityStatus::kDownloadIgnored);
@@ -264,7 +269,8 @@ struct InsecureDownloadData {
         download_source == DownloadSource::OFFLINE_PAGE ||
         download_source == DownloadSource::INTERNAL_API ||
         download_source == DownloadSource::EXTENSION_API ||
-        download_source == DownloadSource::EXTENSION_INSTALLER) {
+        download_source == DownloadSource::EXTENSION_INSTALLER ||
+        allow_insecure_downloads) {
       is_insecure_download_ = false;
     } else {  // Not ignorable download.
       // TODO(crbug.com/40857867): Add blocking metrics.
@@ -416,8 +422,11 @@ InsecureDownloadStatus GetInsecureDownloadStatusForDownload(
     const download::DownloadItem* item) {
   InsecureDownloadData data(path, item);
 
-  // If the download is fully secure, early abort.
-  if (!data.is_insecure_download_) {
+  static const bool allow_insecure_downloads =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "allow-insecure-downloads");
+  // If the download is fully secure, early abort. Don't nag.
+  if (!data.is_insecure_download_ || allow_insecure_downloads) {
     return InsecureDownloadStatus::SAFE;
   }
 
