diff --git a/android_webview/system_webview_apk_tmpl.gni b/android_webview/system_webview_apk_tmpl.gni
index 5b5ff862fe83d..9f3bc7966617c 100644
--- a/android_webview/system_webview_apk_tmpl.gni
+++ b/android_webview/system_webview_apk_tmpl.gni
@@ -89,6 +89,7 @@ template("system_webview_apk_or_module_tmpl") {
     variables = [
       "force_32_bit=$_force_32_bit",
       "manifest_package=$_manifest_package",
+      "application_label=Thorium WebView",
     ]
     if (_is_trichrome) {
       variables +=
diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 6f1aa2dcd2822..0750b64fd75ef 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and 同
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -148,20 +148,14 @@ if (_is_default_toolchain) {
       "java/res_chromium_base/drawable/themed_app_icon.xml",
       "java/res_chromium_base/mipmap-hdpi/app_icon.png",
       "java/res_chromium_base/mipmap-hdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-hdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-mdpi/app_icon.png",
       "java/res_chromium_base/mipmap-mdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-mdpi/layered_app_icon_background.png",
-      "java/res_chromium_base/mipmap-nodpi/layered_app_icon_foreground.xml",
       "java/res_chromium_base/mipmap-xhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
   }
@@ -1776,20 +1770,14 @@ if (_is_default_toolchain) {
       "java/res_chromium_base/drawable/themed_app_icon.xml",
       "java/res_chromium_base/mipmap-hdpi/app_icon.png",
       "java/res_chromium_base/mipmap-hdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-hdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-mdpi/app_icon.png",
       "java/res_chromium_base/mipmap-mdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-mdpi/layered_app_icon_background.png",
-      "java/res_chromium_base/mipmap-nodpi/layered_app_icon_foreground.xml",
       "java/res_chromium_base/mipmap-xhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
 
@@ -2081,7 +2069,7 @@ if (_is_default_toolchain) {
 
   chrome_public_apk_or_module_tmpl("chrome_public_apk") {
     target_type = "android_apk"
-    apk_name = "ChromePublic"
+    apk_name = "Thorium_Public"
     if (defined(android_app_secondary_abi)) {
       # Ensure 64-bit chrome does not depend on 32-bit things.
       assert_no_deps =
@@ -2097,7 +2085,7 @@ if (_is_default_toolchain) {
   }
 
   chrome_public_bundle("chrome_public_bundle") {
-    bundle_name = "ChromePublic"
+    bundle_name = "Thorium_Public_Bundle"
   }
 
   generate_jni("base_module_jni") {
@@ -2334,7 +2322,7 @@ if (_is_default_toolchain) {
   # scoped, and do not leave lingering state after execution.
   # TODO(crbug.com/40193330): Set clear rules for what tests can be added here.
   chrome_public_test_apk_tmpl("chrome_public_unit_test_apk") {
-    apk_name = "ChromePublicUnitTest"
+    apk_name = "ThoriumPublicUnitTest"
     deps = [
       ":chrome_unit_test_java",
       "//build/config/android/test/resource_overlay:unit_device_javatests",
@@ -2407,7 +2395,7 @@ if (_is_default_toolchain) {
 
   # TODO(crbug.com/40193330): Rename to chrome_public_integration_test_apk
   chrome_public_test_apk_tmpl("chrome_public_test_apk") {
-    apk_name = "ChromePublicTest"
+    apk_name = "ThoriumPublicTest"
     deps = [
       ":chrome_test_apk_template_resources",
       ":chrome_test_java",
diff --git a/chrome/android/chrome_java_resources.gni b/chrome/android/chrome_java_resources.gni
index a13c34ab9fdbc..d8348d20b8335 100644
--- a/chrome/android/chrome_java_resources.gni
+++ b/chrome/android/chrome_java_resources.gni
@@ -53,6 +53,7 @@ chrome_java_resources = [
   "java/res/drawable-hdpi/ic_exit_to_app_white_24dp.png",
   "java/res/drawable-hdpi/ic_file_download_white_24dp.png",
   "java/res/drawable-hdpi/ic_link_grey600_36dp.png",
+  "java/res/drawable-hdpi/ic_most_visited_placeholder.png",
   "java/res/drawable-hdpi/ic_phone_googblue_36dp.png",
   "java/res/drawable-hdpi/ic_photo_camera.png",
   "java/res/drawable-hdpi/ic_place_googblue_36dp.png",
@@ -95,6 +96,7 @@ chrome_java_resources = [
   "java/res/drawable-mdpi/ic_exit_to_app_white_24dp.png",
   "java/res/drawable-mdpi/ic_file_download_white_24dp.png",
   "java/res/drawable-mdpi/ic_link_grey600_36dp.png",
+  "java/res/drawable-mdpi/ic_most_visited_placeholder.png",
   "java/res/drawable-mdpi/ic_phone_googblue_36dp.png",
   "java/res/drawable-mdpi/ic_photo_camera.png",
   "java/res/drawable-mdpi/ic_place_googblue_36dp.png",
@@ -139,6 +141,7 @@ chrome_java_resources = [
   "java/res/drawable-xhdpi/ic_exit_to_app_white_24dp.png",
   "java/res/drawable-xhdpi/ic_file_download_white_24dp.png",
   "java/res/drawable-xhdpi/ic_link_grey600_36dp.png",
+  "java/res/drawable-xhdpi/ic_most_visited_placeholder.png",
   "java/res/drawable-xhdpi/ic_phone_googblue_36dp.png",
   "java/res/drawable-xhdpi/ic_photo_camera.png",
   "java/res/drawable-xhdpi/ic_place_googblue_36dp.png",
@@ -180,6 +183,7 @@ chrome_java_resources = [
   "java/res/drawable-xxhdpi/ic_exit_to_app_white_24dp.png",
   "java/res/drawable-xxhdpi/ic_file_download_white_24dp.png",
   "java/res/drawable-xxhdpi/ic_link_grey600_36dp.png",
+  "java/res/drawable-xxhdpi/ic_most_visited_placeholder.png",
   "java/res/drawable-xxhdpi/ic_phone_googblue_36dp.png",
   "java/res/drawable-xxhdpi/ic_photo_camera.png",
   "java/res/drawable-xxhdpi/ic_place_googblue_36dp.png",
@@ -219,6 +223,7 @@ chrome_java_resources = [
   "java/res/drawable-xxxhdpi/ic_exit_to_app_white_24dp.png",
   "java/res/drawable-xxxhdpi/ic_file_download_white_24dp.png",
   "java/res/drawable-xxxhdpi/ic_link_grey600_36dp.png",
+  "java/res/drawable-xxxhdpi/ic_most_visited_placeholder.png",
   "java/res/drawable-xxxhdpi/ic_phone_googblue_36dp.png",
   "java/res/drawable-xxxhdpi/ic_photo_camera.png",
   "java/res/drawable-xxxhdpi/ic_place_googblue_36dp.png",
diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 9bf07d2470457..26c1c75715df1 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -1,4 +1,4 @@
-# Copyright 2015 The Chromium Authors
+# Copyright 2026 The Chromium Authors and 同
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -25,15 +25,15 @@ declare_args() {
   # Android package name to use when compiling the public chrome targets
   # (chrome_public_apk, trichrome_chrome_bundle, etc. as well as the
   # *_bundle variants). This defaults to the Chromium browser package name.
-  chrome_public_manifest_package = "org.chromium.chrome"
+  chrome_public_manifest_package = "org.chromium.thorium"
   if (use_stable_package_name_for_trichrome) {
     chrome_public_manifest_package += ".stable"
-  } else if (android_channel != "default") {
+  } else if (android_channel != "default" && android_channel != "stable") {
     chrome_public_manifest_package += "." + android_channel
   }
 }
 
-chrome_public_test_manifest_package = "org.chromium.chrome.tests"
+chrome_public_test_manifest_package = "org.chromium.thorium.tests"
 
 default_chrome_public_jinja_variables = [
   "channel=$android_channel",
diff --git a/chrome/android/java/res/drawable/ic_warning_white_24dp.xml b/chrome/android/java/res/drawable/ic_warning_white_24dp.xml
index 6e6325a258830..48a67b1c452b6 100644
--- a/chrome/android/java/res/drawable/ic_warning_white_24dp.xml
+++ b/chrome/android/java/res/drawable/ic_warning_white_24dp.xml
@@ -10,6 +10,6 @@ found in the LICENSE file.
     android:height="24dp"
     android:viewportWidth="960"
     android:viewportHeight="960">
-    <path android:fillColor="@android:color/white"
+    <path android:fillColor="#FBBC04"
         android:pathData="M40,840L480,80L920,840L40,840ZM480,720Q497,720 508.5,708.5Q520,697 520,680Q520,663 508.5,651.5Q497,640 480,640Q463,640 451.5,651.5Q440,663 440,680Q440,697 451.5,708.5Q463,720 480,720ZM440,600L520,600L520,400L440,400L440,600Z"/>
 </vector>
\ No newline at end of file
diff --git a/chrome/android/java/res_base/drawable/ic_launcher_round.xml b/chrome/android/java/res_base/drawable/ic_launcher_round.xml
index 2159da0bd9866..cb5e738e61240 100644
--- a/chrome/android/java/res_base/drawable/ic_launcher_round.xml
+++ b/chrome/android/java/res_base/drawable/ic_launcher_round.xml
@@ -5,8 +5,8 @@ Use of this source code is governed by a BSD-style license that can be
 found in the LICENSE file.
 -->
 
-<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android" >
-    <background android:drawable="@mipmap/layered_app_icon_background"/>
-    <foreground android:drawable="@mipmap/layered_app_icon_foreground"/>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@android:color/white"/>
+    <foreground android:drawable="@mipmap/layered_app_icon"/>
     <monochrome android:drawable="@drawable/themed_app_icon"/>
 </adaptive-icon>
diff --git a/chrome/android/java/res_chromium_base/drawable/themed_app_icon.xml b/chrome/android/java/res_chromium_base/drawable/themed_app_icon.xml
index 32a95bc6ee4d5..58acce75bcfb3 100644
--- a/chrome/android/java/res_chromium_base/drawable/themed_app_icon.xml
+++ b/chrome/android/java/res_chromium_base/drawable/themed_app_icon.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
-Copyright 2022 The Chromium Authors
+Copyright 2026 The Chromium Authors and 同
 Use of this source code is governed by a BSD-style license that can be
 found in the LICENSE file.
 -->
diff --git a/chrome/android/java/res_chromium_base/values/channel_constants.xml b/chrome/android/java/res_chromium_base/values/channel_constants.xml
index cc1f2a30cb3c5..32f97e3bb15d2 100644
--- a/chrome/android/java/res_chromium_base/values/channel_constants.xml
+++ b/chrome/android/java/res_chromium_base/values/channel_constants.xml
@@ -1,14 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
-Copyright 2015 The Chromium Authors
+Copyright 2026 The Chromium Authors and 同
 Use of this source code is governed by a BSD-style license that can be
 found in the LICENSE file.
 -->
 
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
     <!-- The application name displayed to the user. -->
-    <string name="app_name" translatable="false">Chromium</string>
-    <string name="bookmark_widget_title" translatable="false">Chromium bookmarks</string>
-    <string name="search_widget_title" translatable="false">Chromium search</string>
-    <string name="quick_action_search_widget_title" translatable="false">Chromium quick action search</string>
+    <string name="app_name" translatable="false">Thorium</string>
+    <string name="bookmark_widget_title" translatable="false">Thorium Bookmarks</string>
+    <string name="search_widget_title" translatable="false">Thorium Search</string>
+    <string name="quick_action_search_widget_title" translatable="false">Thorium Quick Action Search</string>
 </resources>
