diff --git a/chrome/browser/ui/layout_constants.cc b/chrome/browser/ui/layout_constants.cc
index 0af3486d89f3b..cc4b80966419b 100644
--- a/chrome/browser/ui/layout_constants.cc
+++ b/chrome/browser/ui/layout_constants.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/layout_constants.h"
 
+#include "base/command_line.h"
 #include "base/notreached.h"
 #include "build/build_config.h"
 #include "chrome/browser/ui/tabs/features.h"
@@ -18,6 +19,8 @@
 
 int GetLayoutConstant(LayoutConstant constant) {
   const bool touch_ui = ui::TouchUiController::Get()->touch_ui();
+  static const bool classic_omnibox =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
   const bool thorium_2024 = features::IsThorium2024();
   switch (constant) {
     case LayoutConstant::kAppMenuProfileRowAvatarIconSize:
@@ -54,7 +57,7 @@ int GetLayoutConstant(LayoutConstant constant) {
     case LayoutConstant::kLocationBarChildInternalSpacing:
       return touch_ui ? 10 : 4;
     case LayoutConstant::kLocationBarChildCornerRadius:
-      return thorium_2024 ? 8 : 12;
+      return classic_omnibox ? 4 : (thorium_2024 ? 8 : 12);
     case LayoutConstant::kLocationBarChipIconSize:
       return 16;
     case LayoutConstant::kLocationBarChipPadding:
diff --git a/chrome/browser/ui/views/location_bar/location_bar_view.cc b/chrome/browser/ui/views/location_bar/location_bar_view.cc
index 2e383a06ea35f..449daaedecbea 100644
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -11,6 +11,7 @@
 #include <utility>
 
 #include "base/containers/adapters.h"
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/functional/callback_helpers.h"
@@ -271,7 +272,11 @@ LocationBarView::LocationBarView(Browser* browser,
           }));
       views::FocusRing::Get(this)->SetOutsetFocusRingDisabled(true);
     }
-    views::InstallPillHighlightPathGenerator(this);
+    static const bool classic_omnibox =
+        base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
+    if (!classic_omnibox) {
+      views::InstallPillHighlightPathGenerator(this);
+    }
 
 #if BUILDFLAG(OS_LEVEL_GEOLOCATION_PERMISSION_SUPPORTED)
     if (features::IsOsLevelGeolocationPermissionSupportEnabled()) {
@@ -641,6 +646,11 @@ bool LocationBarView::IsInitialized() const {
 }
 
 int LocationBarView::GetBorderRadius() const {
+  static const bool classic_omnibox =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
+  if (classic_omnibox) {
+    return 4;
+  }
   return ComputeBorderRadius(size());
 }
 
@@ -1626,6 +1636,12 @@ void LocationBarView::RefreshBackground() {
   }
 
   SkColor border_color = SK_ColorTRANSPARENT;
+  static const bool classic_omnibox =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
+  if (classic_omnibox && !is_caret_visible) {
+    border_color = color_provider->GetColor(kColorLocationBarBorderOnMismatch);
+  }
+
 
   // The custom focus border is disabled on macOS to avoid conflicting with
   // native omnibox rendering.
diff --git a/chrome/browser/ui/views/omnibox/omnibox_suggestion_button_row_view.cc b/chrome/browser/ui/views/omnibox/omnibox_suggestion_button_row_view.cc
index 29daf84c09d4e..0cad65bccd004 100644
--- a/chrome/browser/ui/views/omnibox/omnibox_suggestion_button_row_view.cc
+++ b/chrome/browser/ui/views/omnibox/omnibox_suggestion_button_row_view.cc
@@ -6,6 +6,7 @@
 
 #include <algorithm>
 
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/memory/raw_ptr.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
@@ -303,7 +304,10 @@ class OmniboxSuggestionRowButton : public views::MdTextButton {
     SetImageLabelSpacing(8);
     SetCustomPadding(ChromeLayoutProvider::Get()->GetInsetsMetric(
         INSETS_OMNIBOX_PILL_BUTTON));
-    SetCornerRadius(GetLayoutConstant(LayoutConstant::kToolbarCornerRadius));
+    static const bool classic_omnibox =
+        base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
+    static const int kCornerRadius = classic_omnibox ? 4 : 8;
+    SetCornerRadius(kCornerRadius);
 
     auto* const ink_drop = views::InkDrop::Get(this);
     SetAnimationDuration(base::TimeDelta());
diff --git a/ui/views/layout/layout_provider.cc b/ui/views/layout/layout_provider.cc
index faa41ea3b6709..804a78485c98b 100644
--- a/ui/views/layout/layout_provider.cc
+++ b/ui/views/layout/layout_provider.cc
@@ -6,6 +6,7 @@
 
 #include <algorithm>
 
+#include "base/command_line.h"
 #include "base/containers/fixed_flat_map.h"
 #include "base/logging.h"
 #include "ui/base/ui_base_features.h"
@@ -227,6 +228,8 @@ ShapeSysTokens GetShapeSysToken(ShapeContextTokens id) {
 
 int LayoutProvider::GetCornerRadiusMetric(ShapeContextTokens id,
                                           const gfx::Size& size) const {
+  static const bool classic_omnibox =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("classic-omnibox");
   if (features::IsThorium2024()) {
     switch (id) {
       case ShapeContextTokens::kBadgeRadius:
@@ -243,7 +246,7 @@ int LayoutProvider::GetCornerRadiusMetric(ShapeContextTokens id,
       case ShapeContextTokens::kMenuTouchRadius:
         return GetCornerRadiusMetric(Emphasis::kHigh);
       case ShapeContextTokens::kOmniboxExpandedRadius:
-        return 16;
+        return classic_omnibox ? 8 : 16;
       case ShapeContextTokens::kTextfieldRadius:
         return 4;
       case ShapeContextTokens::kContentSeparatorRadius:
