diff --git a/net/dns/dns_client.cc b/net/dns/dns_client.cc
index 1d3c4855111be..331eab0b9838c 100644
--- a/net/dns/dns_client.cc
+++ b/net/dns/dns_client.cc
@@ -10,6 +10,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/logging.h"
@@ -330,13 +331,23 @@ class DnsClientImpl : public DnsClient {
   }
 
  private:
+  const bool disable_thorium_dns_config_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-dns-config");
+
   std::optional<DnsConfig> BuildEffectiveConfig() const {
     DnsConfig config;
-    if (config_overrides_.OverridesEverything()) {
+    // In Thorium it is sufficient to have secure DoH enabled to give the
+    // overrides priority.
+    if (config_overrides_.dns_over_https_config &&
+        config_overrides_.secure_dns_mode && !disable_thorium_dns_config_) {
       config = config_overrides_.ApplyOverrides(DnsConfig());
     } else {
-      if (!system_config_)
+      if (!system_config_) {
+        LOG(WARNING) << "dns_client.cc->BuildEffectiveConfig(): System "
+                        "configuration not set: No system_config_ ";
         return std::nullopt;
+      }
 
       config = config_overrides_.ApplyOverrides(system_config_.value());
     }
@@ -352,8 +363,11 @@ class DnsClientImpl : public DnsClient {
       config.nameservers.clear();
     }
 
-    if (!config.IsValid())
+    if (!config.IsValid()) {
+      DLOG(WARNING)
+          << "dns_client.cc->BuildEffectiveConfig(): Invalid configuration";
       return std::nullopt;
+    }
 
     return config;
   }
