diff --git a/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc b/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc
index 036092b985baa..1a054ff5fe643 100644
--- a/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc
+++ b/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -36,43 +36,6 @@
 #include "content/public/browser/visibility.h"
 #include "content/public/browser/web_contents.h"
 
-namespace {
-
-// Do not show the prompt if "suppress_default_browser_prompt_for_version" in
-// the initial preferences is set to the current version.
-bool ShouldShowDefaultBrowserPromptForCurrentVersion() {
-  const std::string disable_version_string =
-      g_browser_process->local_state()->GetString(
-          prefs::kBrowserSuppressDefaultBrowserPrompt);
-  const base::Version disable_version(disable_version_string);
-  DCHECK(disable_version_string.empty() || disable_version.IsValid());
-  return !(disable_version.IsValid() &&
-           disable_version == version_info::GetVersion());
-}
-
-void OnCheckIsDefaultBrowserFinished(
-    Profile* profile,
-    base::OnceCallback<void(bool)> done_callback,
-    shell_integration::DefaultWebClientState state) {
-  bool did_show_prompt = false;
-  if (state == shell_integration::IS_DEFAULT) {
-    // Notify the user in the future if Chrome ceases to be the user's chosen
-    // default browser.
-    chrome::startup::default_prompt::ResetPromptPrefs(profile);
-  } else if (state == shell_integration::NOT_DEFAULT &&
-             shell_integration::CanSetAsDefaultBrowser() &&
-             ShouldShowDefaultBrowserPromptForCurrentVersion()) {
-    // Only show the prompt if some other program is the user's default browser.
-    // In particular, don't show it if another install mode is default (e.g.,
-    // don't prompt for Chrome Beta if stable Chrome is the default).
-    did_show_prompt =
-        DefaultBrowserPromptManager::GetInstance()->MaybeShowPrompt();
-  }
-  std::move(done_callback).Run(did_show_prompt);
-}
-
-}  // namespace
-
 void RegisterDefaultBrowserPromptPrefs(PrefRegistrySimple* registry) {
   registry->RegisterStringPref(prefs::kBrowserSuppressDefaultBrowserPrompt,
                                std::string());
@@ -119,8 +82,8 @@ void MigrateDefaultBrowserLastDeclinedPref(PrefService* profile_prefs) {
   }
 }
 
-void ShowDefaultBrowserPrompt(Profile* profile,
-                              base::OnceCallback<void(bool)> done_callback) {
+void ShowDefaultBrowserPrompt(Profile*,
+                              base::OnceCallback<void(bool)>) {
   // Do not check if Chrome is the default browser if there is a policy in
   // control of this setting.
   if (g_browser_process->local_state()->IsManagedPreference(
@@ -130,7 +93,5 @@ void ShowDefaultBrowserPrompt(Profile* profile,
     return;
   }
 
-  default_browser::DefaultBrowserManager::From(g_browser_process)
-      ->GetDefaultBrowserState(base::BindOnce(
-          &OnCheckIsDefaultBrowserFinished, profile, std::move(done_callback)));
+  return;
 }
