diff --git a/chrome/browser/ui/startup/google_api_keys_infobar_delegate.cc b/chrome/browser/ui/startup/google_api_keys_infobar_delegate.cc
index 78ba37b940137..0733416f0981c 100644
--- a/chrome/browser/ui/startup/google_api_keys_infobar_delegate.cc
+++ b/chrome/browser/ui/startup/google_api_keys_infobar_delegate.cc
@@ -1,26 +1,17 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
 #include "chrome/browser/ui/startup/google_api_keys_infobar_delegate.h"
 
-#include <memory>
-
-#include "chrome/browser/infobars/confirm_infobar_creator.h"
 #include "chrome/grit/branded_strings.h"
-#include "components/infobars/content/content_infobar_manager.h"
-#include "components/infobars/core/infobar.h"
 #include "components/strings/grit/components_strings.h"
 #include "google_apis/google_api_keys.h"
 #include "ui/base/l10n/l10n_util.h"
 
 // static
 void GoogleApiKeysInfoBarDelegate::Create(
-    infobars::ContentInfoBarManager* infobar_manager) {
-  infobar_manager->AddInfoBar(
-      CreateConfirmInfoBar(std::unique_ptr<ConfirmInfoBarDelegate>(
-          new GoogleApiKeysInfoBarDelegate())));
-}
+    infobars::ContentInfoBarManager*) {}
 
 GoogleApiKeysInfoBarDelegate::GoogleApiKeysInfoBarDelegate() = default;
 
diff --git a/chrome/browser/ui/startup/infobar_utils.cc b/chrome/browser/ui/startup/infobar_utils.cc
index 4907b9a951040..f48cf677c7aee 100644
--- a/chrome/browser/ui/startup/infobar_utils.cc
+++ b/chrome/browser/ui/startup/infobar_utils.cc
@@ -1,4 +1,4 @@
-// Copyright 2021 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -182,20 +182,6 @@ void AddInfoBarsIfNecessary(BrowserWindowInterface* browser,
     ShowBadFlagsPrompt(web_contents);
   }
 
-  infobars::ContentInfoBarManager* infobar_manager =
-      infobars::ContentInfoBarManager::FromWebContents(web_contents);
-
-  if (!google_apis::HasAPIKeyConfigured()) {
-    GoogleApiKeysInfoBarDelegate::Create(infobar_manager);
-  }
-
-  if (ObsoleteSystem::IsObsoleteNowOrSoon()) {
-    PrefService* local_state = g_browser_process->local_state();
-    if (!local_state ||
-        !local_state->GetBoolean(prefs::kSuppressUnsupportedOSWarning)) {
-      ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
-    }
-  }
 
   OSCryptAsyncAvailabilityInfoBarDelegate::MaybeCreate(browser);
 
