diff --git a/chrome/browser/ui/browser_actions.cc b/chrome/browser/ui/browser_actions.cc
index 5be3c74fb1d35..3122bb37ca9bd 100644
--- a/chrome/browser/ui/browser_actions.cc
+++ b/chrome/browser/ui/browser_actions.cc
@@ -11,6 +11,7 @@
 
 #include "base/check_deref.h"
 #include "base/check_op.h"
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/functional/callback_helpers.h"
 #include "base/metrics/user_metrics.h"
@@ -829,6 +830,9 @@ void BrowserActions::InitializeChromeMenuActions() {
   TabStripModel* const tab_strip_model = bwi_->GetTabStripModel();
   BrowserWindowInterface* const bwi = base::to_address(bwi_);
   const bool is_guest_session = profile->IsGuestSession();
+  static const bool disable_thorium_command_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
 
   root_action_item_->AddChild(
       ChromeMenuAction(
@@ -841,8 +845,12 @@ void BrowserActions::InitializeChromeMenuActions() {
               profile),
           kActionNewIncognitoWindow, IDS_NEW_INCOGNITO_WINDOW,
           IDS_NEW_INCOGNITO_WINDOW,
-          features::IsRoundedIconsEnabled() ? kIncognitoIcon
-                                            : kIncognitoRefreshMenuOldIcon)
+          disable_thorium_command_icons
+              ? (features::IsRoundedIconsEnabled() ? kIncognitoIcon
+                                                   : kIncognitoRefreshMenuOldIcon)
+              : (features::IsRoundedIconsEnabled()
+                     ? kIncognitoThoriumIcon
+                     : kIncognitoRefreshMenuThoriumIcon))
           .SetEnabled(IncognitoModePrefs::IsIncognitoAllowed(profile))
           .Build());
 
@@ -997,9 +1005,11 @@ void BrowserActions::InitializeChromeMenuActions() {
                            bwi, is_incognito),
                        kActionClearBrowsingData, IDS_CLEAR_BROWSING_DATA,
                        IDS_CLEAR_BROWSING_DATA,
-                       features::IsRoundedIconsEnabled()
-                           ? kDeleteIcon
-                           : kTrashCanRefreshOldIcon)
+                       disable_thorium_command_icons
+                           ? (features::IsRoundedIconsEnabled()
+                                  ? kDeleteIcon
+                                  : kTrashCanRefreshOldIcon)
+                           : kTrashCanRefreshThoriumIcon)
           .SetEnabled(is_incognito ||
                       (!is_guest_session && !profile->IsSystemProfile()))
           .Build());
@@ -1213,9 +1223,7 @@ void BrowserActions::InitializeChromeMenuActions() {
                 },
                 bwi),
             kActionShowChromeLabs, IDS_CHROMELABS, IDS_CHROMELABS,
-            features::IsRoundedIconsEnabled()   ? kScienceIcon
-            : features::IsRoundedIconsEnabled() ? vector_icons::kScienceIcon
-                                                : kScienceOldIcon)
+            disable_thorium_command_icons ? kScienceIcon : kScienceThoriumIcon)
             .SetVisible(ShouldShowChromeLabsUI(profile))
             .Build());
   }
diff --git a/chrome/browser/ui/content_settings/content_setting_image_model.cc b/chrome/browser/ui/content_settings/content_setting_image_model.cc
index bcdbd0957b0e7..c0eba8e182044 100644
--- a/chrome/browser/ui/content_settings/content_setting_image_model.cc
+++ b/chrome/browser/ui/content_settings/content_setting_image_model.cc
@@ -8,6 +8,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/metrics/field_trial_params.h"
 #include "base/metrics/user_metrics.h"
@@ -382,6 +383,9 @@ const ContentSettingsImageDetails* GetImageDetails(ContentSettingsType type) {
 void GetIconChromeRefresh(ContentSettingsType type,
                           bool blocked,
                           raw_ptr<const gfx::VectorIcon>* icon) {
+  const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   switch (type) {
     case ContentSettingsType::COOKIES:
       *icon = blocked ? &(features::IsRoundedIconsEnabled()
@@ -419,11 +423,17 @@ void GetIconChromeRefresh(ContentSettingsType type,
       return;
     case ContentSettingsType::SOUND:
       *icon = blocked ? &(features::IsRoundedIconsEnabled()
-                              ? vector_icons::kVolumeOffIcon
-                              : vector_icons::kVolumeOffChromeRefreshOldIcon)
+                              ? (disable_thorium_icons
+                                     ? vector_icons::kVolumeOffIcon
+                                     : vector_icons::kVolumeOffThoriumIcon)
+                              : (disable_thorium_icons
+                                     ? vector_icons::kVolumeOffChromeRefreshOldIcon
+                                     : vector_icons::kVolumeOffChromeRefreshThoriumIcon))
                       : &(features::IsRoundedIconsEnabled()
                               ? vector_icons::kVolumeUpIcon
-                              : vector_icons::kVolumeUpChromeRefreshOldIcon);
+                              : (disable_thorium_icons
+                                     ? vector_icons::kVolumeUpChromeRefreshOldIcon
+                                     : vector_icons::kVolumeUpChromeRefreshThoriumIcon));
       return;
     case ContentSettingsType::ADS:
       *icon = blocked ? &(features::IsRoundedIconsEnabled()
diff --git a/chrome/browser/ui/extensions/extensions_toolbar_view_model.cc b/chrome/browser/ui/extensions/extensions_toolbar_view_model.cc
index 2ec1cf51cd517..7676198f2faee 100644
--- a/chrome/browser/ui/extensions/extensions_toolbar_view_model.cc
+++ b/chrome/browser/ui/extensions/extensions_toolbar_view_model.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/extensions/extensions_toolbar_view_model.h"
 
+#include "base/command_line.h"
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
 #include "chrome/browser/extensions/extension_tab_util.h"
@@ -176,9 +177,14 @@ const gfx::VectorIcon& ExtensionsToolbarViewModel::GetToolbarButtonIcon(
     ExtensionsToolbarButtonState state) {
   switch (state) {
     case ExtensionsToolbarButtonState::kDefault:
-      return features::IsRoundedIconsEnabled()
-                 ? vector_icons::kChromeExtensionIcon
-                 : vector_icons::kExtensionChromeRefreshOldIcon;
+      return base::CommandLine::ForCurrentProcess()->HasSwitch(
+                 "disable-thorium-icons")
+                 ? (features::IsRoundedIconsEnabled()
+                        ? vector_icons::kChromeExtensionIcon
+                        : vector_icons::kExtensionChromeRefreshOldIcon)
+                 : (features::IsRoundedIconsEnabled()
+                        ? vector_icons::kChromeExtensionIcon
+                        : vector_icons::kExtensionChromeRefreshThoriumIcon);
     case ExtensionsToolbarButtonState::kAllExtensionsBlocked:
       return features::IsRoundedIconsEnabled()
                  ? vector_icons::kChromeExtensionOffIcon
diff --git a/chrome/browser/ui/tabs/alert/tab_alert_icon.cc b/chrome/browser/ui/tabs/alert/tab_alert_icon.cc
index 08e5d23de4982..4504a91a608e4 100644
--- a/chrome/browser/ui/tabs/alert/tab_alert_icon.cc
+++ b/chrome/browser/ui/tabs/alert/tab_alert_icon.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/tabs/alert/tab_alert_icon.h"
 
+#include "base/command_line.h"
 #include "chrome/app/vector_icons/vector_icons.h"
 #include "chrome/browser/actor/resources/grit/actor_browser_resources.h"
 #include "chrome/browser/glic/browser_ui/glic_vector_icon_manager.h"
@@ -66,15 +67,24 @@ ui::ColorId GetAlertIndicatorColor(TabAlert state,
 }
 
 const gfx::VectorIcon& GetAlertIcon(TabAlert alert_state) {
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   switch (alert_state) {
     case TabAlert::kAudioPlaying:
       return features::IsRoundedIconsEnabled()
                  ? vector_icons::kVolumeUpIcon
-                 : vector_icons::kVolumeUpChromeRefreshOldIcon;
+                 : disable_thorium_icons
+                       ? vector_icons::kVolumeUpChromeRefreshOldIcon
+                       : vector_icons::kVolumeUpChromeRefreshThoriumIcon;
     case TabAlert::kAudioMuting:
       return features::IsRoundedIconsEnabled()
-                 ? vector_icons::kVolumeOffIcon
-                 : vector_icons::kVolumeOffChromeRefreshOldIcon;
+                 ? (disable_thorium_icons
+                        ? vector_icons::kVolumeOffIcon
+                        : vector_icons::kVolumeOffThoriumIcon)
+                 : (disable_thorium_icons
+                        ? vector_icons::kVolumeOffChromeRefreshOldIcon
+                        : vector_icons::kVolumeOffChromeRefreshThoriumIcon);
     case TabAlert::kMediaRecording:
     case TabAlert::kAudioRecording:
     case TabAlert::kVideoRecording:
diff --git a/chrome/browser/ui/tabs/tab_menu_model.cc b/chrome/browser/ui/tabs/tab_menu_model.cc
index d59e95f5a5ebe..9b525ad02e9ea 100644
--- a/chrome/browser/ui/tabs/tab_menu_model.cc
+++ b/chrome/browser/ui/tabs/tab_menu_model.cc
@@ -446,15 +446,24 @@ void TabMenuModel::Build(int index) {
                           IDS_TAB_CXMENU_SOUND_UNMUTE_SITE, num_tabs));
 
   if (features::IsMenuSimplificationEnabled()) {
+    static const bool disable_thorium_icons =
+        base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "disable-thorium-icons");
     SetIconForCommandId(
         TabStripModel::CommandToggleSiteMuted,
         ui::ImageModel::FromVectorIcon(
             will_mute ? features::IsRoundedIconsEnabled()
-                            ? vector_icons::kVolumeOffIcon
-                            : vector_icons::kVolumeOffChromeRefreshOldIcon
-            : features::IsRoundedIconsEnabled()
-                ? vector_icons::kVolumeUpIcon
-                : vector_icons::kVolumeUpChromeRefreshOldIcon,
+                            ? (disable_thorium_icons
+                                   ? vector_icons::kVolumeOffIcon
+                                   : vector_icons::kVolumeOffThoriumIcon)
+                            : (disable_thorium_icons
+                                   ? vector_icons::kVolumeOffChromeRefreshOldIcon
+                                   : vector_icons::kVolumeOffChromeRefreshThoriumIcon)
+                       : features::IsRoundedIconsEnabled()
+                           ? vector_icons::kVolumeUpIcon
+                           : (disable_thorium_icons
+                                  ? vector_icons::kVolumeUpChromeRefreshOldIcon
+                                  : vector_icons::kVolumeUpChromeRefreshThoriumIcon),
             ui::kColorMenuIcon, ui::SimpleMenuModel::kDefaultIconSize));
   }
 
diff --git a/chrome/browser/ui/toolbar/app_menu_model.cc b/chrome/browser/ui/toolbar/app_menu_model.cc
index 95ad4098919a8..82f91a9db81d1 100644
--- a/chrome/browser/ui/toolbar/app_menu_model.cc
+++ b/chrome/browser/ui/toolbar/app_menu_model.cc
@@ -771,13 +771,18 @@ void ProfileSubMenuModel::BuildCustomizeProfileRow(Profile* profile) {
 }
 
 void ProfileSubMenuModel::BuildCloseProfileRow(Profile* profile) {
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   AddItemWithIcon(IDC_CLOSE_PROFILE,
                   l10n_util::GetPluralStringFUTF16(IDS_CLOSE_PROFILE,
                                                    CountBrowsersFor(profile)),
                   ui::ImageModel::FromVectorIcon(
-                      features::IsRoundedIconsEnabled()
-                          ? vector_icons::kCloseIcon
-                          : vector_icons::kCloseChromeRefreshOldIcon,
+                      disable_thorium_icons
+                          ? (features::IsRoundedIconsEnabled()
+                                 ? vector_icons::kCloseIcon
+                                 : vector_icons::kCloseChromeRefreshOldIcon)
+                          : vector_icons::kCloseChromeRefreshThoriumIcon,
                       ui::kColorMenuIcon, kDefaultIconSize));
 }
 
@@ -1209,9 +1214,14 @@ void ExtensionsMenuModel::Build(Browser* browser) {
   AddItemWithStringIdAndVectorIcon(
       this, IDC_EXTENSIONS_SUBMENU_MANAGE_EXTENSIONS,
       IDS_EXTENSIONS_SUBMENU_MANAGE_EXTENSIONS_ITEM,
-      features::IsRoundedIconsEnabled()
-          ? vector_icons::kChromeExtensionIcon
-          : vector_icons::kExtensionChromeRefreshOldIcon);
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons")
+          ? (features::IsRoundedIconsEnabled()
+                 ? vector_icons::kChromeExtensionIcon
+                 : vector_icons::kExtensionChromeRefreshOldIcon)
+          : (features::IsRoundedIconsEnabled()
+                 ? vector_icons::kChromeExtensionIcon
+                 : vector_icons::kExtensionChromeRefreshThoriumIcon));
   SetElementIdentifierAt(
       GetIndexOfCommandId(IDC_EXTENSIONS_SUBMENU_MANAGE_EXTENSIONS).value(),
       kManageExtensionsMenuItem);
@@ -2051,6 +2061,9 @@ void AppMenuModel::Build() {
   // Build (and, by extension, Init) should only be called once.
   DCHECK_EQ(0u, GetItemCount());
 
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   if (app_menu_icon_controller_ &&
       app_menu_icon_controller_->GetTypeAndSeverity().type ==
           AppMenuIconController::IconType::kUpgradeNotification) {
@@ -2100,8 +2113,12 @@ void AppMenuModel::Build() {
   if (!browser_->profile()->IsGuestSession()) {
     AddItemWithStringIdAndVectorIcon(
         this, IDC_NEW_INCOGNITO_WINDOW, IDS_NEW_INCOGNITO_WINDOW,
-        features::IsRoundedIconsEnabled() ? kIncognitoIcon
-                                          : kIncognitoRefreshMenuOldIcon);
+        disable_thorium_icons
+            ? (features::IsRoundedIconsEnabled() ? kIncognitoIcon
+                                                 : kIncognitoRefreshMenuOldIcon)
+            : (features::IsRoundedIconsEnabled()
+                   ? kIncognitoThoriumIcon
+                   : kIncognitoRefreshMenuThoriumIcon));
     SetElementIdentifierAt(
         GetIndexOfCommandId(IDC_NEW_INCOGNITO_WINDOW).value(),
         kIncognitoMenuItem);
@@ -2189,9 +2206,13 @@ void AppMenuModel::Build() {
       !extensions::ui_util::HasManageableExtensions(browser_->profile())) {
     AddItemWithStringIdAndVectorIcon(
         this, IDC_FIND_EXTENSIONS, IDS_FIND_EXTENSIONS,
-        features::IsRoundedIconsEnabled()
-            ? vector_icons::kChromeExtensionIcon
-            : vector_icons::kExtensionChromeRefreshOldIcon);
+        disable_thorium_icons
+            ? (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kChromeExtensionIcon
+                   : vector_icons::kExtensionChromeRefreshOldIcon)
+            : (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kChromeExtensionIcon
+                   : vector_icons::kExtensionChromeRefreshThoriumIcon));
     SetElementIdentifierAt(GetIndexOfCommandId(IDC_FIND_EXTENSIONS).value(),
                            ExtensionsMenuModel::kVisitChromeWebStoreMenuItem);
   } else {
@@ -2199,16 +2220,23 @@ void AppMenuModel::Build() {
     AddSubMenuWithStringIdAndVectorIcon(
         this, IDC_EXTENSIONS_SUBMENU, IDS_EXTENSIONS_SUBMENU,
         sub_menus_.back().get(),
-        features::IsRoundedIconsEnabled()
-            ? vector_icons::kChromeExtensionIcon
-            : vector_icons::kExtensionChromeRefreshOldIcon);
+        disable_thorium_icons
+            ? (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kChromeExtensionIcon
+                   : vector_icons::kExtensionChromeRefreshOldIcon)
+            : (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kChromeExtensionIcon
+                   : vector_icons::kExtensionChromeRefreshThoriumIcon));
     SetElementIdentifierAt(GetIndexOfCommandId(IDC_EXTENSIONS_SUBMENU).value(),
                            kExtensionsMenuItem);
   }
 
   AddItemWithStringIdAndVectorIcon(
       this, IDC_CLEAR_BROWSING_DATA, IDS_CLEAR_BROWSING_DATA,
-      kTrashCanRegularIcon);
+      disable_thorium_icons
+          ? (features::IsRoundedIconsEnabled() ? kDeleteIcon
+                                               : kTrashCanRefreshOldIcon)
+          : kTrashCanRegularIcon);
   SetElementIdentifierAt(GetIndexOfCommandId(IDC_CLEAR_BROWSING_DATA).value(),
                          kClearBrowsingDataMenuItem);
 
diff --git a/chrome/browser/ui/views/extensions/extensions_menu_entry_view.cc b/chrome/browser/ui/views/extensions/extensions_menu_entry_view.cc
index 5b4a2131316f4..17bec4a17837f 100644
--- a/chrome/browser/ui/views/extensions/extensions_menu_entry_view.cc
+++ b/chrome/browser/ui/views/extensions/extensions_menu_entry_view.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/extensions/extensions_menu_entry_view.h"
 
+#include "base/command_line.h"
 #include "base/metrics/user_metrics.h"
 #include "base/metrics/user_metrics_action.h"
 #include "chrome/app/vector_icons/vector_icons.h"
@@ -63,12 +64,17 @@ views::Builder<HoverButton> GetSitePermissionsButtonBuilder(
   } else {
     // Add right-aligned arrow icon for non-enterprise extensions when the
     // button is not disabled.
+    static const bool disable_thorium_icons =
+        base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "disable-thorium-icons");
     auto arrow_icon = ui::ImageModel::FromVectorIcon(
-        features::IsThorium2024()
-            ? vector_icons::kSubmenuArrowOldIcon
-            : features::IsRoundedIconsEnabled()
-                ? vector_icons::kKeyboardArrowRightFlippableIcon
-                : vector_icons::kSubmenuArrowChromeRefreshOldIcon,
+        disable_thorium_icons
+            ? (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kKeyboardArrowRightFlippableIcon
+                   : vector_icons::kSubmenuArrowChromeRefreshOldIcon)
+            : features::IsThorium2024()
+                  ? vector_icons::kSubmenuArrowOldIcon
+                  : vector_icons::kSubmenuArrowChromeRefreshThoriumIcon,
         kColorExtensionMenuIcon, small_icon_size);
 
     button_builder.SetHorizontalAlignment(gfx::ALIGN_RIGHT)
diff --git a/chrome/browser/ui/views/global_media_controls/media_toolbar_button_view.cc b/chrome/browser/ui/views/global_media_controls/media_toolbar_button_view.cc
index 5e008d4166e48..d1977d97c5fb7 100644
--- a/chrome/browser/ui/views/global_media_controls/media_toolbar_button_view.cc
+++ b/chrome/browser/ui/views/global_media_controls/media_toolbar_button_view.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/global_media_controls/media_toolbar_button_view.h"
 
+#include "base/command_line.h"
 #include "base/observer_list.h"
 #include "base/strings/pattern.h"
 #include "build/build_config.h"
@@ -58,10 +59,16 @@ MediaToolbarButtonView::MediaToolbarButtonView(
   SetFlipCanvasOnPaintForRTLUI(false);
   SetVectorIcons(features::IsRoundedIconsEnabled()
                      ? kQueueMusicIcon
-                     : kMediaToolbarButtonChromeRefreshOldIcon,
+                     : base::CommandLine::ForCurrentProcess()->HasSwitch(
+                           "disable-thorium-icons")
+                           ? kMediaToolbarButtonChromeRefreshOldIcon
+                           : kMediaToolbarButtonChromeRefreshThoriumIcon,
                  features::IsRoundedIconsEnabled()
                      ? kQueueMusicIcon
-                     : kMediaToolbarButtonTouchOldIcon);
+                     : base::CommandLine::ForCurrentProcess()->HasSwitch(
+                           "disable-thorium-icons")
+                           ? kMediaToolbarButtonTouchOldIcon
+                           : kMediaToolbarButtonTouchThoriumIcon);
   SetTooltipText(
       l10n_util::GetStringUTF16(IDS_GLOBAL_MEDIA_CONTROLS_ICON_TOOLTIP_TEXT));
   GetViewAccessibility().SetHasPopup(ax::mojom::HasPopup::kDialog);
diff --git a/chrome/browser/ui/views/overlay/toggle_mute_button.cc b/chrome/browser/ui/views/overlay/toggle_mute_button.cc
index d2998e3b521b6..9167a1b7fb650 100644
--- a/chrome/browser/ui/views/overlay/toggle_mute_button.cc
+++ b/chrome/browser/ui/views/overlay/toggle_mute_button.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/overlay/toggle_mute_button.h"
 
+#include "base/command_line.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
 #include "chrome/browser/ui/views/overlay/constants.h"
 #include "chrome/grit/generated_resources.h"
@@ -37,13 +38,22 @@ void ToggleMuteButton::UpdateImageAndTooltipText() {
     return;
   }
 
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   const auto& icon = is_muted_
                          ? features::IsRoundedIconsEnabled()
-                               ? vector_icons::kVolumeOffIcon
-                               : vector_icons::kVolumeOffChromeRefreshOldIcon
+                               ? (disable_thorium_icons
+                                      ? vector_icons::kVolumeOffIcon
+                                      : vector_icons::kVolumeOffThoriumIcon)
+                               : (disable_thorium_icons
+                                      ? vector_icons::kVolumeOffChromeRefreshOldIcon
+                                      : vector_icons::kVolumeOffChromeRefreshThoriumIcon)
                      : features::IsRoundedIconsEnabled()
                          ? vector_icons::kVolumeUpIcon
-                         : vector_icons::kVolumeUpChromeRefreshOldIcon;
+                         : (disable_thorium_icons
+                                ? vector_icons::kVolumeUpChromeRefreshOldIcon
+                                : vector_icons::kVolumeUpChromeRefreshThoriumIcon);
 
   auto text = is_muted_ ? IDS_PICTURE_IN_PICTURE_UNMUTE_TEXT
                         : IDS_PICTURE_IN_PICTURE_MUTE_TEXT;
diff --git a/chrome/browser/ui/views/page_info/page_info_main_view.cc b/chrome/browser/ui/views/page_info/page_info_main_view.cc
index eb13023555566..b922957aebc32 100644
--- a/chrome/browser/ui/views/page_info/page_info_main_view.cc
+++ b/chrome/browser/ui/views/page_info/page_info_main_view.cc
@@ -7,6 +7,7 @@
 #include <memory>
 #include <optional>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/i18n/message_formatter.h"
 #include "base/strings/utf_string_conversions.h"
@@ -200,9 +201,12 @@ void PageInfoMainView::SetCookieInfo(const CookiesInfo& cookie_info) {
           base::BindRepeating(&PageInfoNavigationHandler::OpenCookiesPage,
                               base::Unretained(navigation_handler_)),
           PageInfoViewFactory::GetImageModel(
-              features::IsRoundedIconsEnabled()
-                  ? vector_icons::kCookieIcon
-                  : vector_icons::kCookieChromeRefreshOldIcon),
+              base::CommandLine::ForCurrentProcess()->HasSwitch(
+                  "disable-thorium-icons")
+                  ? (features::IsRoundedIconsEnabled()
+                         ? vector_icons::kCookieIcon
+                         : vector_icons::kCookieChromeRefreshOldIcon)
+                  : vector_icons::kCookieChromeRefreshThoriumIcon),
           l10n_util::GetStringUTF16(IDS_PAGE_INFO_COOKIES_HEADER),
           /*subtitle_text=*/std::u16string(),
           PageInfoViewFactory::GetOpenSubpageIcon()));
diff --git a/chrome/browser/ui/views/page_info/page_info_view_factory.cc b/chrome/browser/ui/views/page_info/page_info_view_factory.cc
index 7bca0c5935174..75ff06c111e85 100644
--- a/chrome/browser/ui/views/page_info/page_info_view_factory.cc
+++ b/chrome/browser/ui/views/page_info/page_info_view_factory.cc
@@ -7,6 +7,7 @@
 #include <memory>
 #include <string>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/memory/raw_ptr.h"
 #include "base/strings/utf_string_conversions.h"
@@ -306,6 +307,9 @@ const ui::ImageModel PageInfoViewFactory::GetPermissionIcon(
           ? std::get<ContentSetting>(setting) == CONTENT_SETTING_BLOCK ||
                 std::get<ContentSetting>(setting) == CONTENT_SETTING_ASK
           : info->delegate().IsBlocked(setting);
+  const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
 
   // TODO(crbug.com/335848275): Migrate the icons in 2 steps.
   // 1 - Copy contents of refresh icons into current non-refresh icons.
@@ -449,11 +453,17 @@ const ui::ImageModel PageInfoViewFactory::GetPermissionIcon(
     case ContentSettingsType::SOUND:
       icon = show_blocked_badge
                  ? &(features::IsRoundedIconsEnabled()
-                         ? vector_icons::kVolumeOffIcon
-                         : vector_icons::kVolumeOffChromeRefreshOldIcon)
+                         ? (disable_thorium_icons
+                                ? vector_icons::kVolumeOffIcon
+                                : vector_icons::kVolumeOffThoriumIcon)
+                         : (disable_thorium_icons
+                                ? vector_icons::kVolumeOffChromeRefreshOldIcon
+                                : vector_icons::kVolumeOffChromeRefreshThoriumIcon))
                  : &(features::IsRoundedIconsEnabled()
                          ? vector_icons::kVolumeUpIcon
-                         : vector_icons::kVolumeUpChromeRefreshOldIcon);
+                         : (disable_thorium_icons
+                                ? vector_icons::kVolumeUpChromeRefreshOldIcon
+                                : vector_icons::kVolumeUpChromeRefreshThoriumIcon));
       break;
     case ContentSettingsType::CLIPBOARD_READ_WRITE:
       icon = show_blocked_badge ? &(features::IsRoundedIconsEnabled()
@@ -913,11 +923,19 @@ const ui::ImageModel PageInfoViewFactory::GetOpenSubpageIcon() {
   // GetIconSize() does not work for subpage icons because the default size of
   // features::IsRoundedIconsEnabled() ? vector_icons::kArrowRightFlippableIcon
   // : kSubmenuArrowOldIcon is 8 rather than 16.
-  const int kIconSize = features::IsThorium2024() ? 9 : 20;
+  const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
+  const int kIconSize =
+      disable_thorium_icons ? 20 : features::IsThorium2024() ? 9 : 20;
   return ui::ImageModel::FromVectorIcon(
-      features::IsRoundedIconsEnabled() && !features::IsThorium2024()
-          ? vector_icons::kKeyboardArrowRightFlippableIcon
-          : vector_icons::kSubmenuArrowChromeRefreshOldIcon,
+      disable_thorium_icons
+          ? (features::IsRoundedIconsEnabled()
+                 ? vector_icons::kKeyboardArrowRightFlippableIcon
+                 : vector_icons::kSubmenuArrowChromeRefreshOldIcon)
+          : (features::IsRoundedIconsEnabled() && !features::IsThorium2024()
+                 ? vector_icons::kKeyboardArrowRightFlippableIcon
+                 : vector_icons::kSubmenuArrowChromeRefreshThoriumIcon),
       ui::kColorIcon, kIconSize);
 }
 
diff --git a/chrome/browser/ui/views/profiles/profile_menu_view.cc b/chrome/browser/ui/views/profiles/profile_menu_view.cc
index 990eab66ec0c1..0fd9167999671 100644
--- a/chrome/browser/ui/views/profiles/profile_menu_view.cc
+++ b/chrome/browser/ui/views/profiles/profile_menu_view.cc
@@ -8,6 +8,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "ash/constants/web_app_id_constants.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
@@ -1151,12 +1152,17 @@ void ProfileMenuView::MaybeBuildCloseBrowsersButton() {
     }
   }
 
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   AddFeatureButton(
       l10n_util::GetPluralStringFUTF16(button_title_id, window_count),
       std::move(callback),
-      features::IsRoundedIconsEnabled()
-          ? vector_icons::kCloseIcon
-          : vector_icons::kCloseChromeRefreshOldIcon);
+      disable_thorium_icons
+          ? (features::IsRoundedIconsEnabled()
+                 ? vector_icons::kCloseIcon
+                 : vector_icons::kCloseChromeRefreshOldIcon)
+          : vector_icons::kCloseChromeRefreshThoriumIcon);
 }
 
 void ProfileMenuView::MaybeBuildSignoutButton() {
diff --git a/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc b/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc
index 77612f77f3acb..c4bea2e6332bd 100644
--- a/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc
+++ b/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -6,6 +6,7 @@
 
 #include <set>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
@@ -146,13 +147,19 @@ void BrowserAppMenuButton::UpdateThemeBasedState() {
 }
 
 void BrowserAppMenuButton::UpdateIcon() {
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   const gfx::VectorIcon& icon =
       ui::TouchUiController::Get()->touch_ui()
           ? features::IsRoundedIconsEnabled() ? kMoreVertIcon
                                               : kBrowserToolsTouchOldIcon
-      : features::IsRoundedIconsEnabled() ? kMoreVertIcon
-                                          : kBrowserToolsChromeRefreshOldIcon;
-  const int icon_size = GetIconSize();
+      : features::IsRoundedIconsEnabled()
+          ? kMoreVertIcon
+          : disable_thorium_icons ? kBrowserToolsChromeRefreshOldIcon
+                                  : kBrowserToolsChromeRefreshThoriumIcon;
+  // Fix Thorium hamburger menu size.
+  constexpr int icon_size = 22;
 
   for (auto state : kButtonStates) {
     SkColor icon_color = GetForegroundColor(state);
diff --git a/chrome/browser/ui/views/toolbar/home_button.cc b/chrome/browser/ui/views/toolbar/home_button.cc
index d23ce42d48414..f3994c578a924 100644
--- a/chrome/browser/ui/views/toolbar/home_button.cc
+++ b/chrome/browser/ui/views/toolbar/home_button.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -7,6 +7,7 @@
 #include <memory>
 #include <string_view>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/functional/callback_helpers.h"
@@ -153,9 +154,14 @@ HomeButton::HomeButton(BrowserWindowInterface* browser_window_interface,
   SetProperty(views::kElementIdentifierKey, kToolbarHomeButtonElementId);
   SetTriggerableEventFlags(ui::EF_LEFT_MOUSE_BUTTON |
                            ui::EF_MIDDLE_MOUSE_BUTTON);
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   SetVectorIcons(features::IsRoundedIconsEnabled()
                      ? kHomeIcon
-                     : kNavigateHomeChromeRefreshOldIcon,
+                     : disable_thorium_icons
+                           ? kNavigateHomeChromeRefreshOldIcon
+                           : kNavigateHomeChromeRefreshThoriumIcon,
                  features::IsRoundedIconsEnabled() ? kHomeIcon
                                                    : kNavigateHomeTouchOldIcon);
   SetTooltipText(l10n_util::GetStringUTF16(IDS_TOOLTIP_HOME));
diff --git a/chrome/browser/ui/views/toolbar/reload_button.cc b/chrome/browser/ui/views/toolbar/reload_button.cc
index a2ed2085675b2..7e7270382e950 100644
--- a/chrome/browser/ui/views/toolbar/reload_button.cc
+++ b/chrome/browser/ui/views/toolbar/reload_button.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -6,6 +6,7 @@
 
 #include <stddef.h>
 
+#include "base/command_line.h"
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
 #include "chrome/app/chrome_command_ids.h"
@@ -78,13 +79,19 @@ ReloadButton::ReloadButton(
       command_updater_(command_updater),
       reload_icon_(features::IsRoundedIconsEnabled()
                        ? vector_icons::kRefreshIcon
-                       : vector_icons::kReloadChromeRefreshOldIcon),
+                       : base::CommandLine::ForCurrentProcess()->HasSwitch(
+                             "disable-thorium-icons")
+                             ? vector_icons::kReloadChromeRefreshOldIcon
+                             : vector_icons::kReloadChromeRefreshThoriumIcon),
       reload_touch_icon_(features::IsRoundedIconsEnabled()
                              ? kRefreshIcon
                              : kReloadTouchOldIcon),
       stop_icon_(features::IsRoundedIconsEnabled()
                      ? kCloseIcon
-                     : kNavigateStopChromeRefreshOldIcon),
+                     : base::CommandLine::ForCurrentProcess()->HasSwitch(
+                           "disable-thorium-icons")
+                           ? kNavigateStopChromeRefreshOldIcon
+                           : kNavigateStopChromeRefreshThoriumIcon),
       stop_touch_icon_(features::IsRoundedIconsEnabled()
                            ? kCloseIcon
                            : kNavigateStopTouchOldIcon),
diff --git a/chrome/browser/ui/views/toolbar/toolbar_controller.cc b/chrome/browser/ui/views/toolbar/toolbar_controller.cc
index 1122043b3895c..0a995e6aeac69 100644
--- a/chrome/browser/ui/views/toolbar/toolbar_controller.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_controller.cc
@@ -10,6 +10,7 @@
 #include <variant>
 #include <vector>
 
+#include "base/command_line.h"
 #include "base/containers/adapters.h"
 #include "base/metrics/user_metrics.h"
 #include "base/metrics/user_metrics_action.h"
@@ -236,6 +237,9 @@ ToolbarController::~ToolbarController() {
 std::vector<ToolbarController::ResponsiveElementInfo>
 ToolbarController::GetDefaultResponsiveElements(Browser* browser) {
   bool is_incognito = browser->profile()->IsIncognitoProfile();
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons");
   // TODO(crbug.com/40912482): Fill in observed identifier.
   // Order matters because it should match overflow menu order top to bottom.
   std::vector<ToolbarController::ResponsiveElementInfo> elements = {
@@ -317,19 +321,26 @@ ToolbarController::GetDefaultResponsiveElements(Browser* browser) {
                IDS_OVERFLOW_MENU_ITEM_TEXT_MEDIA_CONTROLS,
                &(features::IsRoundedIconsEnabled()
                      ? kQueueMusicIcon
-                     : kMediaToolbarButtonChromeRefreshOldIcon),
+                     : disable_thorium_icons
+                           ? kMediaToolbarButtonChromeRefreshOldIcon
+                           : kMediaToolbarButtonChromeRefreshThoriumIcon),
                kToolbarMediaButtonElementId, kToolbarMediaBubbleElementId),
            /*is_section_end=*/true),
        ToolbarController::ResponsiveElementInfo(
            ToolbarController::ElementIdInfo(
                kToolbarAvatarButtonElementId,
                IDS_OVERFLOW_MENU_ITEM_TEXT_PROFILE,
-               is_incognito ? (&(features::IsRoundedIconsEnabled()
-                                     ? kIncognitoIcon
-                                     : kIncognitoRefreshMenuOldIcon))
-                            : (&(features::IsRoundedIconsEnabled()
-                                     ? kAccountCircleIcon
-                                     : kUserAccountAvatarRefreshOldIcon)),
+               is_incognito
+                   ? (&(disable_thorium_icons
+                            ? (features::IsRoundedIconsEnabled()
+                                   ? kIncognitoIcon
+                                   : kIncognitoRefreshMenuOldIcon)
+                            : (features::IsRoundedIconsEnabled()
+                                   ? kIncognitoThoriumIcon
+                                   : kIncognitoRefreshMenuThoriumIcon)))
+                   : (&(features::IsRoundedIconsEnabled()
+                            ? kAccountCircleIcon
+                            : kUserAccountAvatarRefreshOldIcon)),
                kToolbarAvatarButtonElementId, kToolbarAvatarBubbleElementId),
            /*is_section_end=*/false)});
 
diff --git a/ui/views/bubble/bubble_frame_view.cc b/ui/views/bubble/bubble_frame_view.cc
index dc2df50dd75f8..f5aa0de7661c5 100644
--- a/ui/views/bubble/bubble_frame_view.cc
+++ b/ui/views/bubble/bubble_frame_view.cc
@@ -7,6 +7,7 @@
 #include <algorithm>
 #include <optional>
 
+#include "base/command_line.h"
 #include "base/check.h"
 #include "build/build_config.h"
 #include "components/vector_icons/vector_icons.h"
@@ -178,9 +179,13 @@ std::unique_ptr<Label> BubbleFrameView::CreateDefaultTitleLabel(
 std::unique_ptr<Button> BubbleFrameView::CreateCloseButton(
     Button::PressedCallback callback) {
   auto close_button = CreateVectorImageButtonWithNativeTheme(
-      std::move(callback), features::IsRoundedIconsEnabled()
-                               ? vector_icons::kCloseIcon
-                               : vector_icons::kCloseChromeRefreshOldIcon);
+      std::move(callback),
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons")
+          ? (features::IsRoundedIconsEnabled()
+                 ? vector_icons::kCloseIcon
+                 : vector_icons::kCloseChromeRefreshOldIcon)
+          : vector_icons::kCloseChromeRefreshThoriumIcon);
   close_button->SetTooltipText(l10n_util::GetStringUTF16(IDS_APP_CLOSE));
   close_button->GetViewAccessibility().SetName(
       l10n_util::GetStringUTF16(IDS_APP_CLOSE));
diff --git a/ui/views/controls/button/md_text_button_with_down_arrow.cc b/ui/views/controls/button/md_text_button_with_down_arrow.cc
index 4171cd3e9380a..887f198b5e94f 100644
--- a/ui/views/controls/button/md_text_button_with_down_arrow.cc
+++ b/ui/views/controls/button/md_text_button_with_down_arrow.cc
@@ -6,6 +6,7 @@
 
 #include <utility>
 
+#include "base/command_line.h"
 #include "ui/base/metadata/metadata_impl_macros.h"
 #include "ui/base/ui_base_features.h"
 #include "ui/gfx/color_utils.h"
@@ -57,8 +58,11 @@ void MdTextButtonWithDownArrow::StateChanged(ButtonState old_state) {
 void MdTextButtonWithDownArrow::SetDropArrowImage() {
   SkColor drop_arrow_color = label()->GetEnabledColor();
   auto drop_arrow_image = ui::ImageModel::FromVectorIcon(
-      features::IsRoundedIconsEnabled() ? kArrowDropDownIcon
-                                        : kArrowDropDownOldIcon,
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-thorium-icons")
+          ? (features::IsRoundedIconsEnabled() ? kArrowDropDownIcon
+                                               : kArrowDropDownOldIcon)
+          : kArrowDropDownThoriumIcon,
       drop_arrow_color, kDropdownArrowSize);
   SetImageModel(Button::STATE_NORMAL, drop_arrow_image);
 }
diff --git a/ui/views/controls/menu/menu_item_view.cc b/ui/views/controls/menu/menu_item_view.cc
index 559b43a5ee39d..d580c90c6cfa5 100644
--- a/ui/views/controls/menu/menu_item_view.cc
+++ b/ui/views/controls/menu/menu_item_view.cc
@@ -15,6 +15,7 @@
 #include <utility>
 
 #include "base/auto_reset.h"
+#include "base/command_line.h"
 #include "base/containers/adapters.h"
 #include "base/i18n/case_conversion.h"
 #include "base/strings/utf_string_conversions.h"
@@ -1651,12 +1652,17 @@ void MenuItemView::UpdateSelectionBasedState(bool paint_as_selected) {
   last_paint_as_selected_ = paint_as_selected;
   const Colors colors = CalculateColors(paint_as_selected);
   if (submenu_arrow_image_view_) {
+    static const bool disable_thorium_icons =
+        base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "disable-thorium-icons");
     const gfx::VectorIcon& submenu_arrow_icon =
-        features::IsThorium2024()
-            ? vector_icons::kSubmenuArrowOldIcon
-            : features::IsRoundedIconsEnabled()
-                  ? vector_icons::kKeyboardArrowRightFlippableIcon
-                  : vector_icons::kSubmenuArrowChromeRefreshOldIcon;
+        disable_thorium_icons
+            ? (features::IsRoundedIconsEnabled()
+                   ? vector_icons::kKeyboardArrowRightFlippableIcon
+                   : vector_icons::kSubmenuArrowChromeRefreshOldIcon)
+            : features::IsThorium2024()
+                  ? vector_icons::kSubmenuArrowOldIcon
+                  : vector_icons::kSubmenuArrowChromeRefreshThoriumIcon;
     submenu_arrow_image_view_->SetImage(
         ui::ImageModel::FromVectorIcon(submenu_arrow_icon, colors.icon_color));
   }
