diff --git a/components/dom_distiller/core/css/distilledpage_common.css b/components/dom_distiller/core/css/distilledpage_common.css
index 570369d1febfe..f80502ac392de 100644
--- a/components/dom_distiller/core/css/distilledpage_common.css
+++ b/components/dom_distiller/core/css/distilledpage_common.css
@@ -1,4 +1,4 @@
-/* Copyright 2014 The Chromium Authors
+/* Copyright 2026 The Chromium Authors and 同
  * Use of this source code is governed by a BSD-style license that can be
  * found in the LICENSE file. */
 
@@ -26,6 +26,12 @@
   box-sizing: border-box;
 }
 
+/* Material Icons rendering fix */
+.material-icons {
+  font-family: Material Icons;
+  font-style: normal !important;
+}
+
 /* Remove all margins and padding from certain element and add word wrap. */
 
 blockquote,
@@ -282,7 +288,7 @@ video::-webkit-media-controls-fullscreen-button {
 /* Iframe sizing. */
 .youtubeContainer {
   height: 0;
-  /* This is the perecnt height of a standard HD video. */
+  /* This is the percent height of a standard HD video. */
   padding-bottom: 56.25%;
   position: relative;
   width: 100%;
diff --git a/components/dom_distiller/core/dom_distiller_features.cc b/components/dom_distiller/core/dom_distiller_features.cc
index b6fc52b3ee56f..a5b10bc954e81 100644
--- a/components/dom_distiller/core/dom_distiller_features.cc
+++ b/components/dom_distiller/core/dom_distiller_features.cc
@@ -1,4 +1,4 @@
-// Copyright 2015 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -21,47 +21,38 @@
 namespace dom_distiller {
 
 bool IsDomDistillerEnabled() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDomDistiller);
+             switches::kEnableDomDistiller) ||
+         kReaderModeDesktop;
 }
 
 bool ShouldStartDistillabilityService() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDistillabilityService);
+             switches::kEnableDistillabilityService) ||
+         kReaderModeDesktop;
 }
 
-BASE_FEATURE(kReaderModeUseReadability, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kReaderModeUseReadability, base::FEATURE_DISABLED_BY_DEFAULT);
 
 #if !BUILDFLAG(IS_IOS)
 constexpr base::FeatureParam<bool> kReaderModeUseReadabilityUseDistiller{
     &kReaderModeUseReadability, /*name=*/"use_distiller",
-    /*default_value=*/true};
+    /*default_value=*/false};
 #endif
 constexpr base::FeatureParam<int> kReaderModeUseReadabilityHeuristicMinScore{
     &kReaderModeUseReadability, /*name=*/"heuristic_min_score",
-#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
-    /*default_value=*/50
-#else
-    /*default_value=*/100
-#endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
-};
+    /*default_value=*/50};
 constexpr base::FeatureParam<int>
     kReaderModeUseReadabilityHeuristicMinContentLength{
         &kReaderModeUseReadability, /*name=*/"heuristic_min_content_length",
-#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
-        /*default_value=*/160
-#else
-        /*default_value=*/200
-#endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
-    };
+        /*default_value=*/160};
 constexpr base::FeatureParam<int> kReaderModeUseReadabilityMinContentLength{
     &kReaderModeUseReadability, /*name=*/"min_content_length",
-#if !BUILDFLAG(IS_ANDROID)
-    /*default_value=*/0
-#else
-    /*default_value=*/100
-#endif
-};
+    /*default_value=*/100};
 
 bool ShouldUseReadabilityDistiller() {
 #if BUILDFLAG(IS_IOS)
@@ -98,9 +89,9 @@ BASE_FEATURE(kReaderModeBlurTransitionAnimation,
 BASE_FEATURE(kReaderModeDelayBottomSheetPeek,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kReaderModeDistillInApp, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kReaderModeDistillInApp, base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kReaderModeImprovements, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kReaderModeImprovements, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kReaderModeToggleLinks, base::FEATURE_DISABLED_BY_DEFAULT);
 
