diff --git a/chrome/browser/ui/views/location_bar/location_bar_view.cc b/chrome/browser/ui/views/location_bar/location_bar_view.cc
index 24c769cf019fe..5060aa43da14a 100644
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -567,6 +567,10 @@ void LocationBarView::Init() {
                                 PageActionIconType::kAiMode);
   }
 
+  if (browser_ && sharing_hub::HasPageAction(profile_, is_popup_mode_)) {
+    params.types_enabled.push_back(PageActionIconType::kSharingHub);
+  }
+
   if (browser_ && !is_popup_mode_) {
     params.types_enabled.push_back(PageActionIconType::kBookmarkStar);
   }
