diff --git a/components/download/public/common/download_features.cc b/components/download/public/common/download_features.cc
index 8dd8fd7e0f167..dbc8d14b4ae43 100644
--- a/components/download/public/common/download_features.cc
+++ b/components/download/public/common/download_features.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -13,7 +13,7 @@ BASE_FEATURE(kParallelDownloading,
 #if BUILDFLAG(IS_ANDROID)
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
-             base::FEATURE_DISABLED_BY_DEFAULT
+             base::FEATURE_ENABLED_BY_DEFAULT
 #endif
 );
 
