diff --git a/ffmpeg_generated.gni b/ffmpeg_generated.gni
index 45019563e5..42d57267a5 100644
--- a/ffmpeg_generated.gni
+++ b/ffmpeg_generated.gni
@@ -372,20 +372,22 @@ if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
     "libavcodec/x86/fmtconvert_init.c",
   ]
 
-  ffmpeg_asm_sources += [
-    "libavcodec/x86/bswapdsp.asm",
-    "libavcodec/x86/hevc/add_res.asm",
-    "libavcodec/x86/hevc/deblock.asm",
-    "libavcodec/x86/hevc/dequant.asm",
-    "libavcodec/x86/hevc/idct.asm",
-    "libavcodec/x86/hevc/mc.asm",
-    "libavcodec/x86/hevc/sao.asm",
-    "libavcodec/x86/hevc/sao_10bit.asm",
-    "libavcodec/x86/h26x/h2656_inter.asm",
-    "libavcodec/x86/ac3dsp.asm",
-    "libavcodec/x86/ac3dsp_downmix.asm",
-    "libavcodec/x86/fmtconvert.asm",
-  ]
+  if (!(is_android && current_cpu == "x86")) {
+    ffmpeg_asm_sources += [
+      "libavcodec/x86/bswapdsp.asm",
+      "libavcodec/x86/hevc/add_res.asm",
+      "libavcodec/x86/hevc/deblock.asm",
+      "libavcodec/x86/hevc/dequant.asm",
+      "libavcodec/x86/hevc/idct.asm",
+      "libavcodec/x86/hevc/mc.asm",
+      "libavcodec/x86/hevc/sao.asm",
+      "libavcodec/x86/hevc/sao_10bit.asm",
+      "libavcodec/x86/h26x/h2656_inter.asm",
+      "libavcodec/x86/ac3dsp.asm",
+      "libavcodec/x86/ac3dsp_downmix.asm",
+      "libavcodec/x86/fmtconvert.asm",
+    ]
+  }
 
   ffmpeg_c_sources += [
     "libavcodec/x86/autorename_libavcodec_x86_aacpsdsp_init.c",
