diff --git a/extensions/common/api/declarative_net_request.webidl b/extensions/common/api/declarative_net_request.webidl
index 13e9fc03677b4..ef6b963297ccd 100644
--- a/extensions/common/api/declarative_net_request.webidl
+++ b/extensions/common/api/declarative_net_request.webidl
@@ -878,28 +878,29 @@ interface DeclarativeNetRequest {
   static Promise<TestMatchOutcomeResult> testMatchOutcome(
       TestMatchRequestDetails request);
 
+  // Increase MV3 limits in Thorium.
   // The minimum number of static rules guaranteed to an extension across its
   // enabled static rulesets. Any rules above this limit will count towards
   // the <a href="#global-static-rule-limit">global static rule limit</a>.
-  const long GUARANTEED_MINIMUM_STATIC_RULES = 30000;
+  const long GUARANTEED_MINIMUM_STATIC_RULES = 60000;
 
   // The maximum number of combined dynamic and session scoped rules an
   // extension can add.
   [nodoc, deprecated="There is no longer a combined limit. See $(ref:MAX_NUMBER_OF_DYNAMIC_RULES) and $(ref:MAX_NUMBER_OF_SESSION_RULES)."]
-  const long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES = 5000;
+  const long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES = 10000;
 
   // The maximum number of dynamic rules that an extension can add.
-  const long MAX_NUMBER_OF_DYNAMIC_RULES = 30000;
+  const long MAX_NUMBER_OF_DYNAMIC_RULES = 60000;
 
   // The maximum number of "unsafe" dynamic rules that an extension can add.
-  const long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES = 5000;
+  const long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES = 10000;
 
   // The maximum number of session scoped rules that an extension can add.
-  const long MAX_NUMBER_OF_SESSION_RULES = 5000;
+  const long MAX_NUMBER_OF_SESSION_RULES = 10000;
 
   // The maximum number of "unsafe" session scoped rules that an extension can
   // add.
-  const long MAX_NUMBER_OF_UNSAFE_SESSION_RULES = 5000;
+  const long MAX_NUMBER_OF_UNSAFE_SESSION_RULES = 10000;
 
   // Time interval within which <code>MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL
   // getMatchedRules</code> calls can be made, specified in minutes.
@@ -910,20 +911,20 @@ interface DeclarativeNetRequest {
 
   // The number of times <code>getMatchedRules</code> can be called within a
   // period of <code>GETMATCHEDRULES_QUOTA_INTERVAL</code>.
-  const long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL = 20;
+  const long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL = 25;
 
   // The maximum number of regular expression rules that an extension can
   // add. This limit is evaluated separately for the set of dynamic rules and
   // those specified in the rule resources file.
-  const long MAX_NUMBER_OF_REGEX_RULES = 1000;
+  const long MAX_NUMBER_OF_REGEX_RULES = 2000;
 
   // The maximum number of static <code>Rulesets</code> an extension can
   // specify as part of the <code>"rule_resources"</code> manifest key.
-  const long MAX_NUMBER_OF_STATIC_RULESETS = 100;
+  const long MAX_NUMBER_OF_STATIC_RULESETS = 256;
 
   // The maximum number of static <code>Rulesets</code> an extension can
   // enable at any one time.
-  const long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS = 50;
+  const long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS = 128;
 
   // Ruleset ID for the dynamic rules added by the extension.
   [StringValue="_dynamic"]
