diff --git a/chrome/browser/obsolete_system/obsolete_system_linux.cc b/chrome/browser/obsolete_system/obsolete_system_linux.cc
index fcd51c9f79c80..c4af4b5a7b22e 100644
--- a/chrome/browser/obsolete_system/obsolete_system_linux.cc
+++ b/chrome/browser/obsolete_system/obsolete_system_linux.cc
@@ -1,4 +1,4 @@
-// Copyright 2022 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -12,7 +12,8 @@
 const char kLinuxObsoleteUrl[] =
     "https://support.google.com/chrome/?p=unsupported_linux";
 
-// This list contains the most recently obsoleted distros.
+// This list contains the most recently obsoleted distros according to
+// `kLinuxObsoleteUrl`.
 const char* const kObsoleteDistros[] = {
     // Prevent clang-format from adding multiple entries per line.
     // clang-format off
@@ -30,7 +31,7 @@ const char* const kObsoleteDistros[] = {
     "Fedora 38",
     "Ubuntu 14.04",
     "Ubuntu 16.04",
-    "Ubuntu 18.04",
+    //"Ubuntu 18.04",
     "openSUSE Leap 15.0",
     "openSUSE Leap 15.1",
     "openSUSE Leap 15.2",
