diff --git a/ui/base/x/x11_util.cc b/ui/base/x/x11_util.cc
index cf5b14d73376b..a345171bd9c05 100644
--- a/ui/base/x/x11_util.cc
+++ b/ui/base/x/x11_util.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -365,14 +365,9 @@ bool GetCustomFramePrefDefault() {
     return false;
   }
 
-  ui::WindowManagerName wm = GuessWindowManager();
-  // If we don't know which WM is active, conservatively disable custom frames.
-  if (wm == WM_OTHER || wm == WM_UNNAMED) {
-    return false;
-  }
-
-  // Stacking WMs should use custom frames.
-  return !IsWmTiling(wm);
+  // Never default to using the custom title bar. Thorium should integrate with
+  // the desktop environment instead of using a special frame by default.
+  return false;
 }
 
 bool IsWmTiling(WindowManagerName window_manager) {
