diff --git a/tools/clang/scripts/build.py b/tools/clang/scripts/build.py
index 372440de944ac..44a9b8a4ba237 100755
--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -73,6 +73,8 @@ BUG_REPORT_URL = ('https://crbug.com in the Tools>LLVM component,'
 LIBXML2_VERSION = 'libxml2-v2.9.12'
 ZSTD_VERSION = 'zstd-1.5.5'
 
+MIMALLOC_URL = 'https://github.com/microsoft/mimalloc/archive/refs/tags/v3.1.4.tar.gz'
+
 win_sdk_dir = None
 def GetWinSDKDir():
   """Get the location of the current SDK."""
@@ -165,7 +167,7 @@ def CheckoutGitRepo(name, git_url, commit, dir):
         ['git', 'diff-index', '--exit-code', 'HEAD'], fail_hard=False)
         and RunCommand(['git', 'fetch'], fail_hard=False)
         and RunCommand(['git', 'checkout', commit], fail_hard=False)
-        and RunCommand(['git', 'clean', '-f'], fail_hard=False)):
+        and RunCommand(['git', 'clean', '-ffd'], fail_hard=False)):
       return
 
     # If we can't use the current repo, delete it.
@@ -390,49 +392,54 @@ def BuildLibXml2(cc, cxx, cmake_sysroot, mac_deployment_target):
   # WindowsManifestMerger.
   # Also enable WITH_THREADS, else libxml doesn't compile on Linux.
   cmake_args = [
-          '-DCMAKE_C_COMPILER=' + cc,
-          '-DCMAKE_CXX_COMPILER=' + cxx,
-          '-DCMAKE_BUILD_TYPE=Release',
-          '-DCMAKE_INSTALL_PREFIX=install',
-          '-DCMAKE_INSTALL_LIBDIR=lib',
-          '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
-          '-DBUILD_SHARED_LIBS=OFF',
-          '-DLIBXML2_WITH_C14N=OFF',
-          '-DLIBXML2_WITH_CATALOG=OFF',
-          '-DLIBXML2_WITH_DEBUG=OFF',
-          '-DLIBXML2_WITH_DOCB=OFF',
-          '-DLIBXML2_WITH_FTP=OFF',
-          '-DLIBXML2_WITH_HTML=OFF',
-          '-DLIBXML2_WITH_HTTP=OFF',
-          '-DLIBXML2_WITH_ICONV=OFF',
-          '-DLIBXML2_WITH_ICU=OFF',
-          '-DLIBXML2_WITH_ISO8859X=OFF',
-          '-DLIBXML2_WITH_LEGACY=OFF',
-          '-DLIBXML2_WITH_LZMA=OFF',
-          '-DLIBXML2_WITH_MEM_DEBUG=OFF',
-          '-DLIBXML2_WITH_MODULES=OFF',
-          '-DLIBXML2_WITH_OUTPUT=ON',
-          '-DLIBXML2_WITH_PATTERN=OFF',
-          '-DLIBXML2_WITH_PROGRAMS=OFF',
-          '-DLIBXML2_WITH_PUSH=OFF',
-          '-DLIBXML2_WITH_PYTHON=OFF',
-          '-DLIBXML2_WITH_READER=OFF',
-          '-DLIBXML2_WITH_REGEXPS=OFF',
-          '-DLIBXML2_WITH_RUN_DEBUG=OFF',
-          '-DLIBXML2_WITH_SAX1=OFF',
-          '-DLIBXML2_WITH_SCHEMAS=OFF',
-          '-DLIBXML2_WITH_SCHEMATRON=OFF',
-          '-DLIBXML2_WITH_TESTS=OFF',
-          '-DLIBXML2_WITH_THREADS=ON',
-          '-DLIBXML2_WITH_THREAD_ALLOC=OFF',
-          '-DLIBXML2_WITH_TREE=ON',
-          '-DLIBXML2_WITH_VALID=OFF',
-          '-DLIBXML2_WITH_WRITER=OFF',
-          '-DLIBXML2_WITH_XINCLUDE=OFF',
-          '-DLIBXML2_WITH_XPATH=OFF',
-          '-DLIBXML2_WITH_XPTR=OFF',
-          '-DLIBXML2_WITH_ZLIB=OFF',
-      ]
+      '-DCMAKE_C_COMPILER=' + cc,
+      '-DCMAKE_CXX_COMPILER=' + cxx,
+      '-DCMAKE_BUILD_TYPE=Release',
+      '-DCMAKE_INSTALL_PREFIX=install',
+      '-DCMAKE_INSTALL_LIBDIR=lib',
+      '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
+      '-DBUILD_SHARED_LIBS=OFF',
+      '-DCMAKE_C_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DLIBXML2_WITH_C14N=OFF',
+      '-DLIBXML2_WITH_CATALOG=OFF',
+      '-DLIBXML2_WITH_DEBUG=OFF',
+      '-DLIBXML2_WITH_DOCB=OFF',
+      '-DLIBXML2_WITH_FTP=OFF',
+      '-DLIBXML2_WITH_HTML=OFF',
+      '-DLIBXML2_WITH_HTTP=OFF',
+      '-DLIBXML2_WITH_ICONV=OFF',
+      '-DLIBXML2_WITH_ICU=OFF',
+      '-DLIBXML2_WITH_ISO8859X=OFF',
+      '-DLIBXML2_WITH_LEGACY=OFF',
+      '-DLIBXML2_WITH_LZMA=OFF',
+      '-DLIBXML2_WITH_MEM_DEBUG=OFF',
+      '-DLIBXML2_WITH_MODULES=OFF',
+      '-DLIBXML2_WITH_OUTPUT=ON',
+      '-DLIBXML2_WITH_PATTERN=OFF',
+      '-DLIBXML2_WITH_PROGRAMS=OFF',
+      '-DLIBXML2_WITH_PUSH=OFF',
+      '-DLIBXML2_WITH_PYTHON=OFF',
+      '-DLIBXML2_WITH_READER=OFF',
+      '-DLIBXML2_WITH_REGEXPS=OFF',
+      '-DLIBXML2_WITH_RUN_DEBUG=OFF',
+      '-DLIBXML2_WITH_SAX1=OFF',
+      '-DLIBXML2_WITH_SCHEMAS=OFF',
+      '-DLIBXML2_WITH_SCHEMATRON=OFF',
+      '-DLIBXML2_WITH_TESTS=OFF',
+      '-DLIBXML2_WITH_THREADS=ON',
+      '-DLIBXML2_WITH_THREAD_ALLOC=OFF',
+      '-DLIBXML2_WITH_TREE=ON',
+      '-DLIBXML2_WITH_VALID=OFF',
+      '-DLIBXML2_WITH_WRITER=OFF',
+      '-DLIBXML2_WITH_XINCLUDE=OFF',
+      '-DLIBXML2_WITH_XPATH=OFF',
+      '-DLIBXML2_WITH_XPTR=OFF',
+      '-DLIBXML2_WITH_ZLIB=OFF',
+  ]
   if cmake_sysroot:
     cmake_args.append('-DCMAKE_SYSROOT=' + cmake_sysroot)
   if sys.platform == 'darwin':
@@ -461,6 +468,55 @@ def BuildLibXml2(cc, cxx, cmake_sysroot, mac_deployment_target):
   return extra_cmake_flags, extra_cflags
 
 
+class LibMimallocDirs:
+  """
+  The set of directories where mimalloc is located.
+
+  Includes the diractories where the source is unpacked, where it is built,
+  and installed.
+  """
+
+  def __init__(self):
+    self.unzip_dir = LLVM_BUILD_TOOLS_DIR
+    self.src_dir = os.path.join(self.unzip_dir, 'mimalloc-3.1.4')
+    self.build_dir = os.path.join(self.src_dir, 'build')
+
+
+def BuildLibMimalloc():
+  """Download and build mimalloc"""
+
+  dirs = LibMimallocDirs()
+  if os.path.exists(dirs.src_dir):
+    RmTree(dirs.src_dir)
+  DownloadAndUnpack(MIMALLOC_URL, dirs.unzip_dir)
+  os.mkdir(dirs.build_dir)
+  os.chdir(dirs.build_dir)
+
+  RunCommand([
+      'cmake',
+      '-GNinja',
+      '-DCMAKE_BUILD_TYPE=Release',
+      '-DCMAKE_C_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DMI_OVERRIDE=ON',
+      '-DMI_SHOW_ERRORS=OFF',
+      '..',
+  ],
+             setenv=True)
+  RunCommand(['ninja'], setenv=True)
+
+  MIMALLOC_LIB = os.path.abspath(
+      os.path.join(LLVM_DIR, '..',
+                   'llvm-build-tools/mimalloc-3.1.4')) + '/build/mimalloc.o'
+
+  mimalloc_ldflags = ['' + MIMALLOC_LIB]
+
+  return mimalloc_ldflags
+
+
 class ZStdDirs:
   """
   The set of directories where zstd is located.
@@ -501,14 +557,22 @@ def BuildZStd(cc, cxx, cmake_sysroot, mac_deployment_target):
   os.chdir(dirs.build_dir)
 
   cmake_args = [
-          '-DCMAKE_C_COMPILER=' + cc,
-          '-DCMAKE_CXX_COMPILER=' + cxx,
-          '-DCMAKE_BUILD_TYPE=Release',
-          '-DCMAKE_INSTALL_PREFIX=install',
-          '-DCMAKE_INSTALL_LIBDIR=lib',
-          '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
-          '-DZSTD_BUILD_SHARED=OFF',
-      ]
+      '-DCMAKE_C_COMPILER=' + cc,
+      '-DCMAKE_CXX_COMPILER=' + cxx,
+      '-DCMAKE_BUILD_TYPE=Release',
+      '-DCMAKE_INSTALL_PREFIX=install',
+      '-DCMAKE_INSTALL_LIBDIR=lib',
+      '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
+      '-DZSTD_BUILD_SHARED=OFF',
+      '-DCMAKE_C_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang'),
+      '-DCMAKE_CXX_COMPILER=' +
+      os.path.join(PINNED_CLANG_DIR, 'bin', 'clang++'),
+      '-DCMAKE_C_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -mtune=skylake -fdata-sections -ffunction-sections',
+      '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+      '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all',
+  ]
   if cmake_sysroot:
     cmake_args.append('-DCMAKE_SYSROOT=' + cmake_sysroot)
   if sys.platform == 'darwin':
@@ -757,21 +821,9 @@ def main():
                       action='store_true',
                       help='use the cmake from PATH instead of downloading '
                       'and using prebuilt cmake binaries')
-  parser.add_argument('--tf-path',
-                      help='path to python tensorflow pip package. '
-                      'Used for embedding an MLGO model')
-  parser.add_argument(
-      '--with-ml-inliner-model',
-      help='path to MLGO inliner model to embed. Setting to '
-      '\'default\', will download an official model which was '
-      'trained for Chrome on Android',
-      default='default' if sys.platform.startswith('linux') else '')
   parser.add_argument('--with-android', type=gn_arg, nargs='?', const=True,
                       help='build the Android ASan runtime (linux only)',
                       default=sys.platform.startswith('linux'))
-  parser.add_argument('--pic',
-                      action='store_true',
-                      help='Uses PIC when building LLVM')
   parser.add_argument('--with-fuchsia',
                       type=gn_arg,
                       nargs='?',
@@ -789,10 +841,6 @@ def main():
                       metavar='WRAPPER',
                       help='Use a compiler wrapper (like ccache) to build the '
                       'stage 1 compiler')
-  parser.add_argument('--without-zstd',
-                      dest='with_zstd',
-                      action='store_false',
-                      help='Disable zstd in the build')
   parser.add_argument('--preserve-gcs-signature',
                       action='store_true',
                       help='By default, this script removes gcs hash files '
@@ -805,8 +853,14 @@ def main():
 
   global CLANG_REVISION, PACKAGE_VERSION, LLVM_BUILD_DIR, STAMP_FILE, FORCE_HEAD_REVISION_FILE
 
-  if (args.pgo or args.thinlto) and not args.bootstrap:
-    print('--pgo/--thinlto requires --bootstrap')
+  if not sys.platform.startswith('linux'):
+    print('Customized build script only works with Linux')
+    return 1
+  if not args.bootstrap:
+    print('Customized build script requires --bootstrap')
+    return 1
+  if args.skip_checkout:
+    print('Customized build script does not support --skip-checkout')
     return 1
   if args.with_android and not os.path.exists(ANDROID_NDK_DIR):
     print('Android NDK not found at ' + ANDROID_NDK_DIR)
@@ -831,10 +885,6 @@ def main():
     print('for general Fuchsia build instructions.')
     return 1
 
-  if args.with_ml_inliner_model and not sys.platform.startswith('linux'):
-    print('--with-ml-inliner-model only supports linux hosts')
-    return 1
-
   # Don't buffer stdout, so that print statements are immediately flushed.
   # LLVM tests print output without newlines, so with buffering they won't be
   # immediately printed.
@@ -909,12 +959,18 @@ def main():
   cc, cxx, lld = None, None, None
   cmake_sysroot = None
 
-  cflags = []
-  cxxflags = []
-  ldflags = []
+  cflags = [
+      '-O3 -mtune=skylake -fdata-sections -ffunction-sections -pthread'
+  ]
+  cxxflags = [
+      '-O3 -mtune=skylake -fdata-sections -ffunction-sections -pthread'
+  ]
+  ldflags = [
+      '-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -pthread'
+  ]
 
   targets = 'AArch64;ARM;LoongArch;Mips;PowerPC;RISCV;SystemZ;WebAssembly;X86'
-  projects = 'clang;lld'
+  projects = 'clang;lld;polly'
   if not args.no_tools:
     projects += ';clang-tools-extra'
   if args.bolt:
@@ -927,7 +983,6 @@ def main():
     runtimes = 'compiler-rt'
 
   pic_default = sys.platform == 'win32'
-  pic_mode = 'ON' if args.pic or pic_default else 'OFF'
 
   base_cmake_args = [
       '-GNinja',
@@ -937,7 +992,7 @@ def main():
       f'-DLLVM_ENABLE_PROJECTS={projects}',
       f'-DLLVM_ENABLE_RUNTIMES={runtimes}',
       f'-DLLVM_TARGETS_TO_BUILD={targets}',
-      f'-DLLVM_ENABLE_PIC={pic_mode}',
+      f'-DLLVM_ENABLE_PIC=ON',
       '-DLLVM_ENABLE_Z3_SOLVER=OFF',
       '-DCLANG_PLUGIN_SUPPORT=OFF',
       '-DCLANG_ENABLE_STATIC_ANALYZER=OFF',
@@ -961,7 +1016,8 @@ def main():
       # `bootstrap` tool:
       # https://github.com/rust-lang/rust/blob/021861aea8de20c76c7411eb8ada7e8235e3d9b5/src/bootstrap/src/core/build_steps/llvm.rs#L348
       '-DLLVM_INSTALL_UTILS=ON',
-      '-DLLVM_ENABLE_ZSTD=%s' % ('ON' if args.with_zstd else 'OFF'),
+      '-DLLVM_ENABLE_ZSTD=ON',
+      '-DLLVM_ENABLE_BINDINGS=OFF'
   ]
 
   if sys.platform == 'darwin':
@@ -1029,6 +1085,10 @@ def main():
 
   deployment_target = '10.12'
 
+  #Build mimalloc
+  libmimalloc_ldflags = BuildLibMimalloc()
+  ldflags += libmimalloc_ldflags
+
   # Statically link libxml2 to make lld-link not require mt.exe on Windows,
   # and to make sure lld-link output on other platforms is identical to
   # lld-link on Windows (for cross-builds).
@@ -1039,14 +1099,13 @@ def main():
   cflags += libxml_cflags
   cxxflags += libxml_cflags
 
-  if args.with_zstd:
-    # Statically link zstd to make lld support zstd compression for debug info.
-    with timer.time('zstd build'):
-      zstd_cmake_args, zstd_cflags = BuildZStd(cc, cxx, cmake_sysroot,
-                                               deployment_target)
-    base_cmake_args += zstd_cmake_args
-    cflags += zstd_cflags
-    cxxflags += zstd_cflags
+  # Statically link zstd to make lld support zstd compression for debug info.
+  with timer.time('zstd build'):
+    zstd_cmake_args, zstd_cflags = BuildZStd(cc, cxx, cmake_sysroot,
+                                             deployment_target)
+  base_cmake_args += zstd_cmake_args
+  cflags += zstd_cflags
+  cxxflags += zstd_cflags
 
   lit_excludes = []
   if sys.platform.startswith('linux'):
@@ -1112,7 +1171,7 @@ def main():
       bootstrap_targets += ';ARM;AArch64'
     bootstrap_args = base_cmake_args + compiler_wrapper_cmake_args + [
         '-DLLVM_TARGETS_TO_BUILD=' + bootstrap_targets,
-        '-DLLVM_ENABLE_PROJECTS=clang;lld',
+        '-DLLVM_ENABLE_PROJECTS=clang;lld;polly',
         '-DLLVM_ENABLE_RUNTIMES=' + ';'.join(runtimes),
         '-DCMAKE_INSTALL_PREFIX=' + LLVM_BOOTSTRAP_INSTALL_DIR,
         '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
@@ -1170,6 +1229,9 @@ def main():
 
     print('Bootstrap compiler installed.')
 
+  cflags += ['-mllvm -polly']
+  cxxflags += ['-mllvm -polly']
+
   if args.pgo:
     print('Building instrumented compiler')
     if os.path.exists(LLVM_INSTRUMENTED_DIR):
@@ -1183,9 +1245,10 @@ def main():
     cxxflags += ['-Wno-backend-plugin']
 
     instrument_args = base_cmake_args + [
-        '-DLLVM_ENABLE_PROJECTS=clang',
-        '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-        '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
+        '-DLLVM_ENABLE_PROJECTS=clang;polly',
+        '-DCMAKE_C_FLAGS= -mllvm -vp-counters-per-site=3 ' + ' '.join(cflags),
+        '-DCMAKE_CXX_FLAGS= -mllvm -vp-counters-per-site=3 ' +
+        ' '.join(cxxflags),
         '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
         '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
         '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
@@ -1230,9 +1293,15 @@ def main():
       training_source = 'pgo_training-3.ii'
       with open(training_source, 'wb') as f:
         DownloadUrl(CDS_URL + '/' + training_source, f)
-      train_cmd = [os.path.join(LLVM_INSTRUMENTED_DIR, 'bin', 'clang++'),
-                  '-target', 'x86_64-unknown-unknown', '-O2', '-g', '-std=c++23',
-                   '-fno-exceptions', '-fno-rtti', '-w', '-c', training_source]
+      train_cmd = [
+          os.path.join(LLVM_INSTRUMENTED_DIR, 'bin',
+                       'clang++'), '-target', 'x86_64-unknown-unknown', '-O3',
+          '-g', '-std=c++23', '-mtune=skylake',
+          '-fmerge-all-constants', '-fdata-sections', '-ffunction-sections',
+          '-fno-exceptions', '-fno-rtti', '-mllvm', '-polly', '-mllvm',
+          '-polly-process-unprofitable', '-Rpass-analysis=polly', '-w', '-c',
+          training_source
+      ]
       if sys.platform == 'darwin':
         train_cmd.extend(['-isysroot', isysroot])
       RunCommand(train_cmd, setenv=True)
@@ -1281,13 +1350,18 @@ def main():
   if cxx is not None: base_cmake_args.append('-DCMAKE_CXX_COMPILER=' + cxx)
   if lld is not None: base_cmake_args.append('-DCMAKE_LINKER=' + lld)
   final_install_dir = args.install_dir if args.install_dir else LLVM_BUILD_DIR
+  tf_path = subprocess.check_output(
+      ['vpython3', os.path.join(THIS_DIR, 'get_tensorflow.py')],
+      universal_newlines=True).rstrip()
+  base_cmake_args += [
+      '-DTENSORFLOW_AOT_PATH=%s' % tf_path,
+  ]
   cmake_args = base_cmake_args + [
-      '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-      '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
-      '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
-      '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
-      '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
       '-DCMAKE_INSTALL_PREFIX=' + final_install_dir,
+      '-DLLVM_INLINER_MODEL_CURRENT_URL=https://commondatastorage.googleapis.com/chromium-browser-clang/tools/mlgo_model3.tgz',
+      '-DLLVM_INLINER_MODEL_PATH=download',
+      '-DLLVM_RAEVICT_MODEL_CURRENT_URL=https://github.com/google/ml-compiler-opt/releases/download/regalloc-evict-v1.1/model.zip',
+      '-DLLVM_RAEVICT_MODEL_PATH=download',
   ]
   if not args.no_tools:
     cmake_args.extend([
@@ -1296,10 +1370,66 @@ def main():
         os.path.join(CHROMIUM_DIR, 'tools', 'clang'),
         '-DCHROMIUM_TOOLS=%s' % ';'.join(chrome_tools)
     ])
-  if args.pgo:
+  if args.pgo and not args.thinlto:
     cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
-  if args.thinlto:
+    cmake_args.append(
+        '-DCMAKE_C_FLAGS=-mllvm -enable-ext-tsp-block-placement ' +
+        ' '.join(cflags))
+    cmake_args.append(
+        '-DCMAKE_CXX_FLAGS=-mllvm -enable-ext-tsp-block-placement ' +
+        ' '.join(cxxflags))
+    cmake_args.append(
+        '-DCMAKE_EXE_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' +
+        ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' +
+        ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_MODULE_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' +
+        ' '.join(ldflags))
+  elif args.thinlto and not args.pgo:
     cmake_args.append('-DLLVM_ENABLE_LTO=Thin')
+    cmake_args.append('-DLLVM_PARALLEL_LINK_JOBS=16')
+    cmake_args.append(
+        '-DCMAKE_C_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' +
+        ' '.join(cflags))
+    cmake_args.append(
+        '-DCMAKE_CXX_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit '
+        + ' '.join(cxxflags))
+    cmake_args.append(
+        '-DCMAKE_EXE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_SHARED_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_MODULE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+  elif args.thinlto and args.pgo:
+    cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
+    cmake_args.append('-DLLVM_ENABLE_LTO=Thin')
+    cmake_args.append('-DLLVM_PARALLEL_LINK_JOBS=16')
+    cmake_args.append(
+        '-DCMAKE_C_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' +
+        ' '.join(cflags))
+    cmake_args.append(
+        '-DCMAKE_CXX_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit '
+        + ' '.join(cxxflags))
+    cmake_args.append(
+        '-DCMAKE_EXE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_SHARED_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+    cmake_args.append(
+        '-DCMAKE_MODULE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly '
+        + ' '.join(ldflags))
+  else:
+    cmake_args.append('-DCMAKE_C_FLAGS= ' + ' '.join(cflags))
+    cmake_args.append('-DCMAKE_CXX_FLAGS= ' + ' '.join(cxxflags))
+    cmake_args.append('-DCMAKE_EXE_LINKER_FLAGS= ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_SHARED_LINKER_FLAGS= ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_MODULE_LINKER_FLAGS= ' + ' '.join(ldflags))
   if sys.platform == 'win32':
     cmake_args.append('-DLLVM_ENABLE_ZLIB=FORCE_ON')
 
@@ -1341,6 +1471,11 @@ def main():
             # TODO(crbug.com/40242553): pass proper flags to i386 tests so they
             # compile correctly
             'LLVM_INCLUDE_TESTS=OFF',
+            'CMAKE_C_FLAGS=-O3 -mtune=skylake -fdata-sections -ffunction-sections -mllvm -polly',
+            'CMAKE_CXX_FLAGS=-O3 -mtune=skylake -fdata-sections -ffunction-sections -mllvm -polly',
+            'CMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly',
+            'CMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly',
+            'CMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly'
         ],
         "profile":
         True,
@@ -1352,12 +1487,17 @@ def main():
             'CMAKE_SYSROOT=%s' % sysroot_amd64,
             # Enable CET IBT so that binaries compiled with
             # -fcf-protection=branch can enable kernel IBT enforcement.
-            'CMAKE_C_FLAGS=-fcf-protection=branch',
-            'CMAKE_CXX_FLAGS=-fcf-protection=branch',
+            'CMAKE_C_FLAGS=-O3 -mtune=skylake -fdata-sections -ffunction-sections -mllvm -polly -fcf-protection=branch',
+            'CMAKE_CXX_FLAGS=-O3 -mtune=skylake -fdata-sections -ffunction-sections -mllvm -polly -fcf-protection=branch',
             'CMAKE_ASM_FLAGS=-fcf-protection=branch',
+            'CMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly',
+            'CMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly',
+            'CMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly',
         ],
-        "profile": True,
-        "sanitizers": True,
+        "profile":
+        True,
+        "sanitizers":
+        True,
     }
     # Using "armv7a-unknown-linux-gnueabhihf" confuses the compiler-rt
     # builtins build, since compiler-rt/cmake/builtin-config-ix.cmake
@@ -1543,32 +1683,6 @@ def main():
           "profile": build_profile
       }
 
-  # Embed MLGO inliner model. If tf_path is not specified, a vpython3 env
-  # will be created which contains the necessary source files for compilation.
-  # MLGO is only officially supported on linux. This condition is checked at
-  # the top of main()
-  if args.with_ml_inliner_model:
-    if args.with_ml_inliner_model == 'default':
-      model_path = ('https://commondatastorage.googleapis.com/'
-                    'chromium-browser-clang/tools/mlgo_model3.tgz')
-    else:
-      model_path = args.with_ml_inliner_model
-    if not args.tf_path:
-      with timer.time('get tensorflow'):
-        tf_path = subprocess.check_output(
-            ['vpython3', os.path.join(THIS_DIR, 'get_tensorflow.py')],
-            universal_newlines=True).rstrip()
-    else:
-      tf_path = args.tf_path
-    print('Embedding MLGO inliner model at %s using Tensorflow at %s' %
-          (model_path, tf_path))
-    cmake_args += [
-        '-DLLVM_INLINER_MODEL_PATH=%s' % model_path,
-        '-DTENSORFLOW_AOT_PATH=%s' % tf_path,
-        # Disable Regalloc model generation since it is unused
-        '-DLLVM_RAEVICT_MODEL_PATH=none'
-    ]
-
   # Convert FOO=BAR CMake flags per triple into
   # -DBUILTINS_$triple_FOO=BAR/-DRUNTIMES_$triple_FOO=BAR and build up
   # -DLLVM_BUILTIN_TARGETS/-DLLVM_RUNTIME_TARGETS.
@@ -1630,7 +1744,7 @@ def main():
     with timer.time('bolt instrument'):
       RunCommand([
           'bin/llvm-bolt', 'bin/clang', '-o', 'bin/clang-bolt.inst',
-          '-instrument', '--instrumentation-file-append-pid',
+          '-lite=false', '-instrument', '--instrumentation-file-append-pid',
           '--instrumentation-file=' +
           os.path.join(bolt_profiles_dir, 'prof.fdata')
       ])
@@ -1643,14 +1757,20 @@ def main():
     # Train by building a part of Clang.
     os.mkdir('bolt-training')
     os.chdir('bolt-training')
+    if args.pgo:
+      cflags += ['-mllvm -enable-ext-tsp-block-placement']
+      cxxflags += ['-mllvm -enable-ext-tsp-block-placement']
+      ldflags += ['-Wl,-z,keep-text-section-prefix']
     bolt_train_cmake_args = base_cmake_args + [
         '-DLLVM_TARGETS_TO_BUILD=X86',
         '-DLLVM_ENABLE_PROJECTS=clang',
-        '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-        '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
-        '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
-        '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
-        '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
+        '-DCMAKE_C_FLAGS=-g -fmerge-all-constants -Rpass-analysis=polly ' +
+        ' '.join(cflags),
+        '-DCMAKE_CXX_FLAGS=-g -fmerge-all-constants -Rpass-analysis=polly ' +
+        ' '.join(cxxflags),
+        '-DCMAKE_EXE_LINKER_FLAGS= ' + ' '.join(ldflags),
+        '-DCMAKE_SHARED_LINKER_FLAGS= ' + ' '.join(ldflags),
+        '-DCMAKE_MODULE_LINKER_FLAGS= ' + ' '.join(ldflags),
         '-DCMAKE_C_COMPILER=' +
         os.path.join(LLVM_BUILD_DIR, 'bin/clang-bolt.inst'),
         '-DCMAKE_CXX_COMPILER=' +
@@ -1659,6 +1779,8 @@ def main():
         os.path.join(LLVM_BUILD_DIR, 'bin/clang-bolt.inst'),
         '-DCMAKE_ASM_COMPILER_ID=Clang',
     ]
+    if args.pgo:
+      bolt_train_cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
     with timer.time('bolt training cmake'):
       RunCommand(['cmake'] + bolt_train_cmake_args +
                  [os.path.join(LLVM_DIR, 'llvm')])
@@ -1677,10 +1799,36 @@ def main():
           'merged.fdata', bolt_profiles_dir
       ])
       RunCommand([
-          'bin/llvm-bolt', 'bin/clang', '-o', 'bin/clang-bolt.opt', '-data',
-          'merged.fdata', '-reorder-blocks=ext-tsp', '-reorder-functions=hfsort+',
-          '-split-functions', '-split-all-cold', '-split-eh', '-dyno-stats',
-          '-icf=1', '-use-gnu-stack', '-use-old-text'
+          'bin/llvm-bolt',
+          'bin/clang',
+          '-o',
+          'bin/clang-bolt.opt',
+          '-data',
+          'merged.fdata',
+          '-reorder-blocks=ext-tsp',
+          '-reorder-functions=cdsort',
+          '-split-functions',
+          '-split-strategy=cdsplit',
+          '-split-all-cold',
+          '-split-eh',
+          '-dyno-stats',
+          '-icf=all',
+          '-use-gnu-stack',
+          '-peepholes=all',
+          '-elim-link-veneers',
+          '-group-stubs',
+          '-align-blocks',
+          '-sctc-mode=heuristic',
+          '-jump-tables=aggressive',
+          '-simplify-rodata-loads',
+          '-tail-duplication=cache',
+          '-indirect-call-promotion=all',
+          '-icp-eliminate-loads',
+          '-hot-data',
+          '-x86-strip-redundant-address-size',
+          '-reorder-functions-use-hot-size',
+          '-reorder-data-algo=funcs',
+          '-lite=false',
       ])
 
     # Overwrite clang, preserving its timestamp so ninja doesn't rebuild it.
@@ -1689,8 +1837,7 @@ def main():
 
   VerifyVersionOfBuiltClangMatchesVERSION()
   VerifyZlibSupport()
-  if args.with_zstd:
-    VerifyZStdSupport()
+  VerifyZStdSupport()
 
   # Run tests.
   if (chrome_tools and (args.run_tests or args.llvm_force_head_revision)):
@@ -1713,6 +1860,11 @@ def main():
                  FORCE_HEAD_REVISION_FILE,
                  preserve_hash_files=args.preserve_gcs_signature)
 
+  RunCommand(
+      ['ln', '--symbolic', '--relative', '-f', 'bin/clang', 'bin/clang++'])
+  RunCommand(
+      ['ln', '--symbolic', '--relative', '-f', 'bin/clang', 'bin/clang-cl'])
+
   print('Clang build was successful.')
 
   return 0
