diff --git a/chrome/app/chrome_command_ids.h b/chrome/app/chrome_command_ids.h
index 0f50c4d73e0cf..204179d257ed3 100644
--- a/chrome/app/chrome_command_ids.h
+++ b/chrome/app/chrome_command_ids.h
@@ -368,6 +368,7 @@
 #define IDC_CONTENT_CONTEXT_OPENLINKINPROFILE 50108
 #define IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP 50109
 #define IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW 50111
+#define IDC_CONTENT_CONTEXT_OPENLINKSAMETAB 50119
 #define IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST 50112
 // Image items.
 #define IDC_CONTENT_CONTEXT_SAVEIMAGEAS 50120
diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 498c452d177f4..b759f76b3e848 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -701,6 +701,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="The name of the Open Link in New Tab command in the content area context menu">
             Open link in new &amp;tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="The name of the Open Link in Current Tab command in the content area context menu">
+            Open link in &amp;current tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="The name of the Open Link in New Window command in the content area context menu">
             Open link in new &amp;window
           </message>
@@ -1040,6 +1043,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="In Title Case: The name of the Open Link in New Tab command in the content area context menu">
             Open Link in New &amp;Tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="In Title Case: The name of the Open Link in Current Tab command in the content area context menu">
+            Open Link in &amp;Current Tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="In Title Case: The name of the Open Link in New Window command in the content area context menu">
             Open Link in New &amp;Window
           </message>
diff --git a/chrome/browser/renderer_context_menu/render_view_context_menu.cc b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
index 055326ebf4431..5b1a90402fd36 100644
--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -616,13 +616,14 @@ const std::map<int, int>& GetIdcToUmaMap(UmaEnumIdLookupType type) {
        {IDC_CONTENT_CONTEXT_LISTEN_TO_THIS_PAGE, 164},
        {IDC_CONTENT_CONTEXT_DICTATION, 165},
        {IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST, 166},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 167},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the RenderViewContextMenuItem enum in
        //     tools/metrics/histograms/metadata/ui/enums.xml.
-       {0, 167}});
+       {0, 168}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:RenderViewContextMenuItem)
 
   // LINT.IfChange(ContextMenuOptionDesktop)
@@ -661,13 +662,14 @@ const std::map<int, int>& GetIdcToUmaMap(UmaEnumIdLookupType type) {
        {IDC_CONTENT_CONTEXT_SEARCHWEBFORNEWTAB, 29},
        {IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW, 31},
        {IDC_CONTENT_CONTEXT_GLICSHAREIMAGE, 32},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 33},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the ContextMenuOptionDesktop enum in
        //     tools/metrics/histograms/metadata/enums.xml.
-       {0, 32}});
+       {0, 33}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:ContextMenuOptionDesktop)
 
   return *(type == UmaEnumIdLookupType::GeneralEnumId ? kGeneralMap
@@ -729,7 +731,8 @@ int FindUMAEnumValueForCommand(int id, UmaEnumIdLookupType type) {
 
 // Returns true if the command id is for opening a link.
 bool IsCommandForOpenLink(int id) {
-  return id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
+  return id == IDC_CONTENT_CONTEXT_OPENLINKSAMETAB ||
+         id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
          id == IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW ||
          id == IDC_CONTENT_CONTEXT_OPENLINKOFFTHERECORD ||
          id == IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW ||
@@ -1935,7 +1938,12 @@ void RenderViewContextMenu::AppendLinkItems() {
       }
     }
 #endif
-
+    if (show_open_in_new_tab) {
+      menu_model_.AddItemWithStringId(
+          IDC_CONTENT_CONTEXT_OPENLINKSAMETAB,
+          in_app ? IDS_CONTENT_CONTEXT_OPENLINKNEWTAB_INAPP
+                 : IDS_CONTENT_CONTEXT_OPENLINKSAMETAB);
+    }
     if (params_.link_url.is_valid()) {
       AppendProtocolHandlerSubMenu();
     }
@@ -3311,6 +3319,7 @@ bool RenderViewContextMenu::IsCommandIdEnabled(int id) const {
 
     case IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP:
     case IDC_CONTENT_CONTEXT_OPENLINKINPROFILE:
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW:
       return navigation_allowed && params_.link_url.is_valid() &&
@@ -3633,6 +3642,30 @@ void RenderViewContextMenu::ExecuteCommand(int id, int event_flags) {
   }
 
   switch (id) {
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB: {
+      WindowOpenDisposition new_tab_disposition =
+          WindowOpenDisposition::CURRENT_TAB;
+      BrowserWindowInterface* browser = nullptr;
+      if (IsInProgressiveWebApp()) {
+        browser = FindNormalBrowser(GetProfile());
+        new_tab_disposition = browser
+                                  ? WindowOpenDisposition::NEW_FOREGROUND_TAB
+                                  : WindowOpenDisposition::NEW_WINDOW;
+      }
+
+      OpenURLParams params = GetOpenURLParamsWithExtraHeaders(
+          params_.link_url, params_.frame_url, params_.frame_origin,
+          new_tab_disposition, ui::PAGE_TRANSITION_LINK,
+          /*extra_headers=*/std::string(), /*started_from_context_menu=*/true);
+
+      if (browser) {
+        browser->OpenURL(params, /*navigation_handle_callback=*/{});
+      } else {
+        source_web_contents_->OpenURL(params,
+                                      /*navigation_handle_callback=*/{});
+      }
+      break;
+    }
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB: {
       WindowOpenDisposition new_tab_disposition =
           WindowOpenDisposition::NEW_BACKGROUND_TAB;
diff --git a/chrome/browser/ui/actions/chrome_action_id.h b/chrome/browser/ui/actions/chrome_action_id.h
index fb93a1bc1e616..cba464deeb599 100644
--- a/chrome/browser/ui/actions/chrome_action_id.h
+++ b/chrome/browser/ui/actions/chrome_action_id.h
@@ -267,6 +267,7 @@
   E(kActionExtensionsContextCustomLast, IDC_EXTENSIONS_CONTEXT_CUSTOM_LAST) \
   /* Context menu items in the render view. */ \
   /* Link items. */ \
+  E(kActionContentContextOpenLinkSameTab, IDC_CONTENT_CONTEXT_OPENLINKSAMETAB) \
   E(kActionContentContextOpenLinkNewTab, IDC_CONTENT_CONTEXT_OPENLINKNEWTAB) \
   E(kActionContentContextOpenLinkNewWindow, \
     IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW) \
diff --git a/tools/metrics/histograms/metadata/ui/enums.xml b/tools/metrics/histograms/metadata/ui/enums.xml
index 0b5754cc68d86..541900f47dee5 100644
--- a/tools/metrics/histograms/metadata/ui/enums.xml
+++ b/tools/metrics/histograms/metadata/ui/enums.xml
@@ -78,6 +78,7 @@ chromium-metrics-reviews@google.com.
   <int value="29" label="Search web for... in new tab"/>
   <int value="31" label="Open link in split view"/>
   <int value="32" label="Create image with Gemini in Chrome"/>
+  <int value="33" label="Open link in this tab"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:ContextMenuOptionDesktop) -->
@@ -367,6 +368,7 @@ chromium-metrics-reviews@google.com.
   <int value="164" label="IDC_CONTENT_CONTEXT_LISTEN_TO_THIS_PAGE"/>
   <int value="165" label="IDC_CONTENT_CONTEXT_DICTATION"/>
   <int value="166" label="IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST"/>
+  <int value="167" label="IDC_CONTENT_CONTEXT_OPENLINKSAMETAB"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:RenderViewContextMenuItem) -->
