diff --git a/net/base/load_flags_list.h b/net/base/load_flags_list.h
index c10df92f30b12..7d3806e1a66b1 100644
--- a/net/base/load_flags_list.h
+++ b/net/base/load_flags_list.h
@@ -1,4 +1,4 @@
-// Copyright 2011 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -123,4 +123,8 @@ LOAD_FLAG(SHOULD_BYPASS_HSTS, 1 << 18)
 // compromised renderer concern is acceptable.
 LOAD_FLAG(IS_MAIN_FRAME_ORIGIN_RECENTLY_ACCESSED, 1 << 19)
 
-// See note at top of file about why adding LoadFlags is often a bad idea.
+// This load will not send Accept-Language or User-Agent headers, and not
+// advertise brotli or zstd encoding.
+// Used to comply with IETF (draft) DNS-over-HTTPS:
+// "Implementors SHOULD NOT set non-essential HTTP headers in DoH client requests."
+LOAD_FLAG(MINIMAL_HEADERS, 1 << 20)
diff --git a/net/dns/dns_http_attempt.cc b/net/dns/dns_http_attempt.cc
index c7afbb1d9071e..a277cf2c23957 100644
--- a/net/dns/dns_http_attempt.cc
+++ b/net/dns/dns_http_attempt.cc
@@ -1,4 +1,4 @@
-// Copyright 2026 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -170,7 +170,7 @@ DnsHTTPAttempt::DnsHTTPAttempt(base::WeakPtr<ResolveContext> resolve_context,
   // avoid deadlock and enable the use of preconfigured IP addresses.
   request_->SetSecureDnsPolicy(SecureDnsPolicy::kBootstrap);
   request_->SetLoadFlags(request_->load_flags() | LOAD_DISABLE_CACHE |
-                         LOAD_BYPASS_PROXY);
+                         LOAD_MINIMAL_HEADERS | LOAD_BYPASS_PROXY);
   request_->set_disallow_credentials();
   request_->set_isolation_info(GetDohIsolationInfo());
 }
diff --git a/net/url_request/url_request_http_job.cc b/net/url_request/url_request_http_job.cc
index fce5645a3f08f..1546695d74f87 100644
--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -532,18 +532,22 @@ void URLRequestHttpJob::OnGotFirstPartySetMetadata(
   // fields in the referrer.
   GURL referrer(request_->referrer());
 
-  // Our consumer should have made sure that this is a safe referrer (e.g. via
-  // URLRequestJob::ComputeReferrerForPolicy).
-  if (referrer.is_valid()) {
-    std::string referer_value = referrer.spec();
-    request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
-                                          referer_value);
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    // Our consumer should have made sure that this is a safe referrer (e.g. via
+    // URLRequestJob::ComputeReferrerForPolicy).
+    if (referrer.is_valid()) {
+      std::string referer_value = referrer.spec();
+      request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
+                                            referer_value);
+    }
   }
 
-  request_info_.extra_headers.SetHeaderIfMissing(
-      HttpRequestHeaders::kUserAgent,
-      http_user_agent_settings_ ? http_user_agent_settings_->GetUserAgent()
-                                : std::string());
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    request_info_.extra_headers.SetHeaderIfMissing(
+        HttpRequestHeaders::kUserAgent,
+        http_user_agent_settings_ ? http_user_agent_settings_->GetUserAgent()
+                                  : std::string());
+  }
 
   AddExtraHeaders();
 
@@ -798,10 +802,13 @@ void URLRequestHttpJob::StartTransactionInternal() {
 void URLRequestHttpJob::AddExtraHeaders() {
   request_info_.extra_headers.SetAcceptEncodingIfMissing(
       request()->url(), request()->accepted_stream_types(),
-      request()->context()->enable_brotli(),
-      request()->context()->enable_zstd());
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) &&
+          request()->context()->enable_brotli(),
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) &&
+          request()->context()->enable_zstd());
 
-  if (http_user_agent_settings_) {
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS) &&
+      http_user_agent_settings_) {
     // Only add default Accept-Language if the request didn't have it
     // specified.
     std::string accept_language =
