diff --git a/chrome/browser/net/default_dns_over_https_config_source.cc b/chrome/browser/net/default_dns_over_https_config_source.cc
index e5a2aadcc85f6..3ce7e0d2088fd 100644
--- a/chrome/browser/net/default_dns_over_https_config_source.cc
+++ b/chrome/browser/net/default_dns_over_https_config_source.cc
@@ -1,4 +1,4 @@
-// Copyright 2024 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -29,7 +29,7 @@ DefaultDnsOverHttpsConfigSource::DefaultDnsOverHttpsConfigSource(
   if (set_up_pref_defaults) {
     local_state->SetDefaultPrefValue(prefs::kDnsOverHttpsMode,
                                      base::Value(SecureDnsConfig::ModeToString(
-                                         net::SecureDnsMode::kAutomatic)));
+                                         net::SecureDnsMode::kSecure)));
   }
 }
 
diff --git a/chrome/browser/net/stub_resolver_config_reader.cc b/chrome/browser/net/stub_resolver_config_reader.cc
index 3d9047c1cf558..444429525ca4a 100644
--- a/chrome/browser/net/stub_resolver_config_reader.cc
+++ b/chrome/browser/net/stub_resolver_config_reader.cc
@@ -1,4 +1,4 @@
-// Copyright 2020 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -20,12 +20,10 @@
 #include "base/values.h"
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
-#include "chrome/browser/browser_process.h"
 #include "chrome/browser/net/default_dns_over_https_config_source.h"
 #include "chrome/browser/net/dns_over_https_config_source.h"
 #include "chrome/browser/net/secure_dns_config.h"
 #include "chrome/browser/net/secure_dns_util.h"
-#include "chrome/browser/policy/chrome_browser_policy_connector.h"
 #include "chrome/common/chrome_features.h"
 #include "chrome/common/pref_names.h"
 #include "components/prefs/pref_registry_simple.h"
@@ -46,20 +44,13 @@
 #endif
 
 #if BUILDFLAG(IS_WIN)
-#include "base/enterprise_util.h"
 #include "base/scoped_native_library.h"
-#include "base/win/win_util.h"
 #include "base/win/windows_version.h"
-#include "chrome/browser/win/parental_controls.h"
 #endif
 
 namespace {
 
 #if BUILDFLAG(IS_WIN)
-bool ShouldDisableDohForWindowsParentalControls() {
-  return GetWinParentalControls().web_filter;
-}
-
 // Defines the base::Feature for controlling the ZTDNS check.
 BASE_FEATURE(kZeroTrustDNS, base::FEATURE_ENABLED_BY_DEFAULT);
 
@@ -238,42 +229,11 @@ void StubResolverConfigReader::UpdateNetworkService(bool record_metrics) {
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForManaged() {
-// This function ignores cloud policies which are loaded on a per-profile basis.
-#if BUILDFLAG(IS_ANDROID)
-  // Check for MDM/management/owner apps. android_has_owner_ is true if either a
-  // device or policy owner app is discovered by
-  // GetAndroidEnterpriseInfoState(). If android_has_owner_ is nullopt, take a
-  // value of false so that we don't disable DoH during the async check.
-
-  // Because Android policies can only be loaded with owner apps this is
-  // sufficient to check for the prescences of policies as well.
-  if (android_has_owner_.value_or(false))
-    return true;
-#elif BUILDFLAG(IS_WIN)
-  // TODO(crbug.com/40229843): What is the correct function to use here? (This
-  // may or may not obsolete the following TODO)
-  // TODO(crbug.com/40223626): For legacy compatibility, this uses
-  // IsEnterpriseDevice() which effectively equates to a domain join check.
-  // Consider whether this should use IsManagedDevice() instead.
-  if (base::win::IsEnrolledToDomain())
-    return true;
-#endif
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
-  if (g_browser_process->browser_policy_connector()->HasMachineLevelPolicies())
-    return true;
-#endif
   return false;
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForParentalControls() {
-  if (parental_controls_testing_override_.has_value())
-    return parental_controls_testing_override_.value();
-
-#if BUILDFLAG(IS_WIN)
-  return ShouldDisableDohForWindowsParentalControls();
-#else
   return false;
-#endif
 }
 
 void StubResolverConfigReader::SetOverrideDnsOverHttpsConfigSource(
