diff --git a/chrome/app/chrome_main_delegate.cc b/chrome/app/chrome_main_delegate.cc
index df8f9455d224f..f0857b4b67eff 100644
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -632,14 +632,13 @@ void InitLogging(const std::string& process_type) {
       *base::CommandLine::ForCurrentProcess();
   logging::InitChromeLogging(command_line, file_state);
 
-#if BUILDFLAG(IS_CHROMEOS)
   // Log the Chrome version for information. Do so at WARNING level as that's
   // the min level on ChromeOS.
   if (process_type.empty()) {
-    LOG(WARNING) << "This is Chrome version " << chrome::kChromeVersion
-                 << " (not a warning)";
+    LOG(WARNING) << "This is " << version_info::GetProductName() << " version: "
+                 << chrome::kChromeVersion << " (not a warning)";
+    LOG(WARNING) << "In Loving Memory of Linda Louise Frick: Grandma, Mother, and Sister.";
   }
-#endif  // BUILDFLAG(IS_CHROMEOS)
 }
 #endif  // !BUILDFLAG(IS_ANDROID)
 
diff --git a/chrome/common/logging_chrome.cc b/chrome/common/logging_chrome.cc
index 2e7ae1d3a591c..6c235fe0a9955 100644
--- a/chrome/common/logging_chrome.cc
+++ b/chrome/common/logging_chrome.cc
@@ -573,7 +573,7 @@ base::FilePath GetLogFileName(const base::CommandLine& command_line) {
 
   // If command line and environment do not provide a log file we can use,
   // fallback to the default.
-  const base::FilePath log_filename(FILE_PATH_LITERAL("chrome_debug.log"));
+  const base::FilePath log_filename(FILE_PATH_LITERAL("thorium_debug.log"));
   base::FilePath log_path;
 
   if (base::PathService::Get(chrome::DIR_LOGS, &log_path)) {
