diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index d210598dfddeb..8a3a6ab7707fa 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -44,7 +44,8 @@ BASE_FEATURE(kAIPageContentMissingSubframesFailSilently,
 
 // Controls the capturing of the Ad-Auction-Signals header, and the maximum
 // allowed Ad-Auction-Signals header value.
-BASE_FEATURE(kAdAuctionSignals, base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled by Alex313031
+BASE_FEATURE(kAdAuctionSignals, base::FEATURE_DISABLED_BY_DEFAULT);
 BASE_FEATURE_PARAM(int,
                    kAdAuctionSignalsMaxSizeBytes,
                    &kAdAuctionSignals,
@@ -97,7 +98,8 @@ BASE_FEATURE_ENUM_PARAM(
     &async_touch_move_throttling_policies);
 
 // Block all MIDI access with the MIDI_SYSEX permission
-BASE_FEATURE(kBlockMidiByDefault, base::FEATURE_ENABLED_BY_DEFAULT);
+// Allow MIDI for Thorium Legacy
+BASE_FEATURE(kBlockMidiByDefault, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kComputePressureRateObfuscationMitigation,
              base::FEATURE_ENABLED_BY_DEFAULT);
@@ -242,11 +244,13 @@ BASE_FEATURE(kBrowsingTopicsBypassIPIsPubliclyRoutableCheck,
 // feature has to be enabled first (i.e. `kBrowsingTopics` is enabled, and,
 // either a valid Origin Trial token exists or `kPrivacySandboxAdsAPIsOverride`
 // is enabled.)
-BASE_FEATURE(kBrowsingTopicsDocumentAPI, base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled by Alex313031
+BASE_FEATURE(kBrowsingTopicsDocumentAPI, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Decoupled with the main `kBrowsingTopics` feature, so it allows us to
 // decouple the server side configs.
-BASE_FEATURE(kBrowsingTopicsParameters, base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled by Alex313031
+BASE_FEATURE(kBrowsingTopicsParameters, base::FEATURE_DISABLED_BY_DEFAULT);
 // The periodic topics calculation interval.
 BASE_FEATURE_PARAM(base::TimeDelta,
                    kBrowsingTopicsTimePeriodPerEpoch,
@@ -836,7 +840,7 @@ BASE_FEATURE(kFetchDestinationJsonCssModules,
 // File handling icons. https://crbug.com/1218213
 BASE_FEATURE(kFileHandlingIcons, base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kFileSystemUrlNavigation, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kFileSystemUrlNavigation, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kFilteringScrollPrediction,
 #if BUILDFLAG(IS_ANDROID)
@@ -855,7 +859,8 @@ BASE_FEATURE_PARAM(std::string,
 
 // See https://github.com/WICG/turtledove/blob/main/FLEDGE.md
 // Enables FLEDGE implementation. See https://crbug.com/1186444.
-BASE_FEATURE(kFledge, base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled in Thorium
+BASE_FEATURE(kFledge, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // See
 // https://github.com/WICG/turtledove/blob/main/FLEDGE_browser_bidding_and_auction_API.md
@@ -2152,7 +2157,8 @@ BASE_FEATURE(kPrerender2EarlyDocumentLifecycleUpdate,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enables the Private Aggregation API.
-BASE_FEATURE(kPrivateAggregationApi, base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled by Alex313031
+BASE_FEATURE(kPrivateAggregationApi, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Selectively allows the JavaScript API to be disabled in just one of the
 // contexts. The Protected Audience param's name has not been updated (from
@@ -2479,7 +2485,7 @@ BASE_FEATURE(
 #if BUILDFLAG(IS_ANDROID)
     // TODO(crbug.com/351953350): Enable this feature on Android as well after
     // confirming that this feature doesn't regress anything.
-    base::FEATURE_DISABLED_BY_DEFAULT
+    base::FEATURE_ENABLED_BY_DEFAULT
 #else
     base::FEATURE_ENABLED_BY_DEFAULT
 #endif
@@ -2501,7 +2507,7 @@ BASE_FEATURE_PARAM(bool,
                    kBoostRenderProcessForLoadingPrioritizePrerendering,
                    &kBoostRenderProcessForLoading,
                    "prioritize_prerendering",
-                   true);
+                   false);
 
 // If true is specified, kBoostRenderProcessForLoading feature only prioritizes
 // the renderer process that is used for prerendering. This is a part of an
