diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 311a3dfebc076..0944c5f707fa3 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and 同
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -641,6 +641,8 @@ source_set("core") {
     "resources_util.cc",
     "shell_integration.cc",
     "startup_data.cc",
+    "thorium_flag_choices.h",
+    "thorium_flag_entries.h",
     "unexpire_flags.cc",
     "unexpire_flags.h",
     "universal_web_contents_observers.cc",
diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
index 65692f66ab989..f888b7904d20b 100644
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -152,6 +152,7 @@
 #include "components/paint_preview/buildflags/buildflags.h"
 #include "components/paint_preview/features/features.h"
 #include "components/password_manager/core/browser/features/password_features.h"
+#include "components/password_manager/core/browser/password_manager_switches.h"
 #include "components/password_manager/core/common/password_manager_features.h"
 #include "components/payments/core/features.h"
 #include "components/performance_manager/public/features.h"
@@ -4694,11 +4695,16 @@ const FeatureEntry::FeatureVariation kCanvasAccessibilityChoices[] = {
 // calculate and verify checksum.
 //
 // When adding a new choice, add it to the end of the list.
+
+// Include Thorium flags.
+#include "chrome/browser/thorium_flag_choices.h"
+
 const FeatureEntry kFeatureEntries[] = {
 // Include generated flags for flag unexpiry; see //docs/flag_expiry.md and
 // //tools/flags/generate_unexpire_flags.py.
 #include "build/chromeos_buildflags.h"
 #include "chrome/browser/unexpire_flags_gen.inc"
+#include "chrome/browser/thorium_flag_entries.h"
     {switches::kEnableBenchmarking, flag_descriptions::kEnableBenchmarkingName,
      flag_descriptions::kEnableBenchmarkingDescription, kOsAll,
      MULTI_VALUE_TYPE(kEnableBenchmarkingChoices)},
