diff --git a/ui/views/BUILD.gn b/ui/views/BUILD.gn
index c44d4794519ac..728edc4b7d299 100644
--- a/ui/views/BUILD.gn
+++ b/ui/views/BUILD.gn
@@ -84,6 +84,10 @@ aggregate_vector_icons("views_vector_icons") {
     "visibility_off.icon",
     "visibility_off_filled.icon",
   ]
+
+  if (is_thorium_build) {
+    sources += [ "thorium/arrow_drop_down_thorium.icon" ]
+  }
 }
 
 generate_allowlist_from_histograms_file("bubble_histograms_variant") {
