diff --git a/chrome/browser/ash/settings/stats_reporting_controller.cc b/chrome/browser/ash/settings/stats_reporting_controller.cc
index 8e3a957236d6c..e28ec9b305b0c 100644
--- a/chrome/browser/ash/settings/stats_reporting_controller.cc
+++ b/chrome/browser/ash/settings/stats_reporting_controller.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -58,9 +58,8 @@ void StatsReportingController::SetEnabled(Profile* profile, bool enabled) {
 }
 
 bool StatsReportingController::IsEnabled() const {
-  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
-  std::optional<base::Value> value = GetValue();
-  return value.has_value() && value->is_bool() && value->GetBool();
+  // Disable Telemetry in ThoriumOS
+  return false;
 }
 
 StatsReportingController::StatsReportingController(PrefService* local_state)
