diff --git a/chrome/app/vector_icons/BUILD.gn b/chrome/app/vector_icons/BUILD.gn
index 7549ff5614f9e..7bb53ae1799df 100644
--- a/chrome/app/vector_icons/BUILD.gn
+++ b/chrome/app/vector_icons/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and 同
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -272,6 +272,7 @@ aggregate_vector_icons("chrome_vector_icons") {
     "qr_code.icon",
     "qr_code_chrome_refresh_old.icon",
     "qrcode_generator_custom.icon",
+    "quick_commands.icon",
     "queue_music.icon",
     "read_later_add_old.icon",
     "reading_list_old.icon",
@@ -367,6 +368,7 @@ aggregate_vector_icons("chrome_vector_icons") {
     "translate_old.icon",
     "trash_can_old.icon",
     "trash_can_refresh_old.icon",
+    "trash_can_regular.icon",
     "travel_explore.icon",
     "travel_explore_old.icon",
     "tv.icon",
@@ -421,6 +423,29 @@ aggregate_vector_icons("chrome_vector_icons") {
     # go/keep-sorted end
   ]
 
+  if (is_thorium_build) {
+    sources += [
+      "thorium/browser_tools_thorium.icon",
+      "thorium/browser_tools_chrome_refresh_thorium.icon",
+      "thorium/chrome_labs_thorium.icon",
+      "thorium/chrome_labs_chrome_refresh_thorium.icon",
+      "thorium/incognito_thorium.icon",
+      "thorium/incognito_refresh_menu_thorium.icon",
+      "thorium/media_toolbar_button_chrome_refresh_thorium.icon",
+      "thorium/media_toolbar_button_touch_thorium.icon",
+      "thorium/navigate_home_thorium.icon",
+      "thorium/navigate_home_chrome_refresh_thorium.icon",
+      "thorium/navigate_stop_chrome_refresh_thorium.icon",
+      "thorium/science_thorium.icon",
+      "thorium/side_panel_left_thorium.icon",
+      "thorium/side_panel_left_chrome_refresh_thorium.icon",
+      "thorium/side_panel_left_touch_thorium.icon",
+      "thorium/side_panel_thorium.icon",
+      "thorium/side_panel_chrome_refresh_thorium.icon",
+      "thorium/side_panel_touch_thorium.icon",
+      "thorium/trash_can_refresh_thorium.icon",
+    ]
+  }
   if (is_mac) {
     sources += [
       "add_circle_filled.icon",
