diff --git a/chrome/browser/ui/toolbar/app_menu_model.cc b/chrome/browser/ui/toolbar/app_menu_model.cc
index 44cc2b51c48a2..95ad4098919a8 100644
--- a/chrome/browser/ui/toolbar/app_menu_model.cc
+++ b/chrome/browser/ui/toolbar/app_menu_model.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and 同
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -1006,9 +1006,7 @@ void HelpMenuModel::Build(Browser* browser) {
     int help_string_id = IDS_HELP_PAGE;
 #endif
     AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
-                                     features::IsRoundedIconsEnabled()
-                                         ? vector_icons::kInfoIcon
-                                         : vector_icons::kInfoRefreshOldIcon);
+                                     vector_icons::kInfoRegularIcon);
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
     if (whats_new::IsEnabled()) {
       AddItemWithStringIdAndVectorIcon(
@@ -2210,8 +2208,7 @@ void AppMenuModel::Build() {
 
   AddItemWithStringIdAndVectorIcon(
       this, IDC_CLEAR_BROWSING_DATA, IDS_CLEAR_BROWSING_DATA,
-      features::IsRoundedIconsEnabled() ? kDeleteIcon
-                                        : kTrashCanRefreshOldIcon);
+      kTrashCanRegularIcon);
   SetElementIdentifierAt(GetIndexOfCommandId(IDC_CLEAR_BROWSING_DATA).value(),
                          kClearBrowsingDataMenuItem);
 
@@ -2309,12 +2306,11 @@ void AppMenuModel::Build() {
                          kHelpMenuItem);
 #else
 #if BUILDFLAG(IS_CHROMEOS)
-  AddItem(IDC_ABOUT, l10n_util::GetStringUTF16(IDS_ABOUT));
+  AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
+                                   vector_icons::kInfoRegularIcon);
 #else
   AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
-                                   features::IsRoundedIconsEnabled()
-                                       ? vector_icons::kInfoIcon
-                                       : vector_icons::kInfoRefreshOldIcon);
+                                   vector_icons::kInfoRegularIcon);
 #endif
 #endif
 
