diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 18417167ff8dd..5705543001704 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1013,7 +1013,7 @@ if (is_win) {
   if (bundle_widevine_cdm) {
     bundle_data("widevine_cdm_library_binaries") {
       sources = [ "$root_out_dir/$widevine_cdm_path/libwidevinecdm.dylib" ]
-      if (enable_widevine_cdm_host_verification) {
+      if (bundle_widevine_cdm_signature) {
         sources +=
             [ "$root_out_dir/$widevine_cdm_path/libwidevinecdm.dylib.sig" ]
       }
